---
layout: project-anvio-main
title: "An analysis and visualization platform for 'omics data"
redirect_from: /projects/anvio/
excerpt: "Anvi'o Project Page"
---

{% include _toc.html %}

{% include _project-anvio-images.html %}

{% include _join-anvio-slack.html %}

# Anvi'o in a nutshell

Anvi'o is an analysis and visualization platform for 'omics data. It brings together many aspects of today's cutting-edge genomic, metagenomic, and metatranscriptomic analysis practices to address a wide array of needs.

Please see the "**Tutorials**" pull-down menu on this page for some example workflows with anvi'o. Anvi'o is a community-driven, GPL-licensed open-source project. You can browse its codebase by clicking [here](http://github.com/meren/anvio).

{:.notice}
Click here to download our poster from the ISME conference in Montreal: [eren-isme-poster-72dpi.png](../../files/eren-isme-poster-72dpi.png)

# Others on anvi'o

[Visit this post]({% post_url anvio/2017-07-12-testimonials %}) to read user testimonials, or share your anvi'o experience with others.

# Citation

Eren AM, Esen ÖC, Quince C, Vineis JH, Morrison HG, Sogin ML, Delmont TO. (2015) [Anvi’o: an advanced analysis and visualization platform for ‘omics data](https://peerj.com/articles/1319/). *PeerJ* **3**:e1319

# Thanks

Anvi'o would not have been possible without the help of many people who directly or indirectly contributed to its development. We thank <a href="https://www.linkedin.com/in/farukuzun" target="_blank">Faruk Uzun</a>, <a href="https://twitter.com/dcankilment" target="_blank">Doğan Can Kilment</a>, <a href="https://github.com/gokmen" target="_blank">Gökmen Göksel</a>, <a href="https://github.com/caglar10ur" target="_blank">S. Çağlar Onur</a>, and <a href="https://github.com/gkmngrgn" target="_blank">Gökmen Görgen</a> for their contributions to the code base, and <a href="http://www.richfox.org/me/" target="_blank">Rich Fox</a> for administering our servers. We thank Inés Martínez for testing anvi’o, and for her valuable suggestions throughout the development of the platform. We thank <a href="https://twitter.com/phyllosphere" target="_blank">Sheri Simmons</a> for suggesting the application of oligotyping to the metagenomic data to characterize single-nucleotide variation. We also thank <a href="https://www.linkedin.com/pub/itai-sharon/0/8a1/93b" target="_blank">Itai Sharon</a>, <a href="https://scholar.google.com/citations?user=NflBi1cAAAAJ" target="_blank">Luis M. Rodridugez-R</a>, <a href="https://scholar.google.com/citations?user=ljGGBFwAAAAJ&hl=en" target="_blank">Will A. Overholt</a>, <a href="http://eoas.fsu.edu/people/faculty/dr-olivia-mason" target="_blank">Olivia U. Mason</a>, <a href="http://www.microbialecology.ca/" target="_blank">Etienne Yergeau</a>, and their colleagues for making valuable datasets available to the science community and for answering our questions.

# All posts anvi'o
