---
layout: post
title: "Anvi'o User Tutorial for Metagenomic Workflow"
excerpt: "Sweets for people who managed to install the platform and want to do genome-resolved metagenomics."
modified: 2016-06-22
tags: []
categories: [anvio]
redirect_from: /2015/05/01/anvio-tutorial/
comments: true
---

{% include _toc.html %}

{% include _project-anvio-version.html %}

{: .notice}
This tutorial is tailored for anvi'o <b>version 2.2.2, or later</b>. You can learn which version you have on your computer by typing `anvi-profile --version` in your terminal. The tutorial for older anvi'o releases from v1 family is [here]({% post_url anvio/2015-05-02-anvio-tutorial-v1 %}) (but **v1** is so 2015, and you should never use it ever again).

The goal of this tutorial is to provide a brief overview of the anvi'o workflow for the analysis of assembly-based shotgun metagenomic data. Throughout this tutorial you will primarily learn about the following topics:

* Process your contigs,

* Profile your metagenomic samples and merge them, 

* Visualize your data, identify and/or refine genome bins interactively, and create summaries of your results.

If we are missing things, or parts of the tutorial is not clear, please let us know, and we will do our best to improve it.

You should also feel free to contact us if you are not sure whether anvi'o is the platform to go after a certain question. 

---

If you are here, you must have already [installed]({% post_url anvio/2016-06-26-installation-v2 %}) the platform (hopefully without much trouble), and have run the infamous ["mini test"]({% post_url anvio/2016-06-26-installation-v2 %}#running-the-mini-test) successfully.

It is always a good idea to stick with stable versions of the platform, as the snapshots from [the codebase](http://github.com/meren/anvio) can be very unstable and/or broken. However we also need people who like to live at the edge, and who would follow the development, test new features, join discussions, and push us to do better.

You probably will run into issues while using anvi'o. We apologize for that in advance. But when that happens, please consider contacting us to give us a chance to address the problem for you. You can post a comment down below, join [the anvi'o discussion group]({% post_url anvio/2015-10-14-anvio-discussion-group %}), or open an <a href="https://github.com/meren/anvio/issues">issue</a>. Regardless of the method to connect, please don't forget to copy-paste the `anvi-interactive -v` output, the operating system you are using, or any other details that may be relevant to the problem.


## Preparation

To run the anvi'o metagenomic workflow, you will need these files:

* __A FASTA file of your contigs__. We shall call it `contigs.fa` throughout this manual. We will assume that `contigs.fa` contains contigs from a co-assembly. However, it may also have been a reference genome from NCBI, a metagenome-assembled genome (MAG), or a bunch of genes you are interested in profiling. Regardless of what it contains, the following steps will not change too much.

* __BAM files__ for your samples. In fact you can use most of what anvi'o offers, including its binning capabilities, [even if you don't have any BAM files]({% post_url anvio/2016-06-06-working-with-contigs-only %}), but since you would be an outlier if that's the case, let's continue with the more conventional scenario. Where you have your contigs, and one or more BAM files.

{:.notice}
This tutorial [starts with BAM files and a FASTA file]({% post_url anvio/2016-06-22-anvio-tutorial-v2 %}#preparation) for your contigs. The reason for that is simple: there are many ways to get your contigs and BAM files for your metagenomes. But we have started implementing a tutorial that describes the workflow _we_ use to generate these files regularly: "[A tutorial on assembly-based metagenomics]({{ site.url }}/tutorials/assembly-based-metagenomics/)". Please feel free to take a look at that one, as well. But this tutorial assumes that you have your BAM files and the FASTA file for your contigs ready.

{:.notice}
To make things easier to follow, we will use three mock samples throughout this tutorial: `SAMPLE-01`, `SAMPLE-02`, and `SAMPLE-03` (in fact, these are subsampled from a human gut metagenome time series). By clicking the following links, you can download the `contigs.fa` and the three BAM files we generated by mapping short reads from each sample to `contigs.fa`: [contigs.fa](https://raw.githubusercontent.com/meren/anvio/master/anvio/tests/sandbox/contigs.fa), [SAMPLE-01-RAW.bam](https://github.com/meren/anvio/raw/master/anvio/tests/sandbox/SAMPLE-01-RAW.bam), [SAMPLE-02-RAW.bam](https://github.com/meren/anvio/raw/master/anvio/tests/sandbox/SAMPLE-02-RAW.bam), [SAMPLE-03-RAW.bam](https://github.com/meren/anvio/raw/master/anvio/tests/sandbox/SAMPLE-03-RAW.bam). Save them into a directory, and run every command in that directory throughout the tutorial.

For the contigs and BAM files for your *real data*, there is one more thing you have to make sure you have: *simple deflines*. Keep reading.

### Take a look at your FASTA file

**Your FASTA file must have simple deflines**, and if it doesn't have simple deflines, **you must fix your FASTA file prior to mapping**. This is necessary, because the names in `contigs.fa` **must** match the names in your BAM files. Unfortunately, different mapping software behave differently when they find a space character, or say a `|` character in your FASTA file, and they proceed to change those characters in arbitrary ways. Therefore it is essential to keep the sequence IDs in your FASTA file **as simple as possible** before mapping. To avoid any problems later, take a look at your deflines prior to mapping now, and remove anything that is not a digit, an ASCII letter, an underscore, or a dash character. Here are some bad deflines:

``` bash
>Contig-123 length:4567 
>Another defline 42
>gi|478446819|gb|JN117275.2|
```

And here are some OK ones:

``` bash
>Contig-123
>Another_defline_42
>gi_478446819_gb_JN117275_2
```

If you have bad deflines, you need to reformat your FASTA file, and do the mapping again (if you have done you mapping already, you can convert your BAM files into SAM files, edit the SAM file to correct deflines, and re-generate your BAM files with proper names, but these kind of error-prone hacks require a lot of attention to make sure you did not introduce a bug early on to your precious data).

### Re-formatting your input FASTA

You can use the following anvi'o script to fix your deflines:

``` bash
$ anvi-script-reformat-fasta contigs.fa -o contigs-fixed.fa -l 0 --simplify-names
```

This script will give you your FASTA file with simplified deflines. If you use the flag `--report-file`, it will also create a TAB-delimited file for you to keep track of which defline in the new file corresponds to which defline in the original file. While you are here, it may also be a good idea to remove some very short contigs from your contigs file for a clean start. If you like that idea, you can run the same command this way to also remove sequences that are shorter than 1,000 nts:

``` bash
$ anvi-script-reformat-fasta contigs.fa -o contigs-fixed.fa -l 1000 --simplify-names
```

Let's just overwrite the `contigs.fa` with `contigs-fixed.fa` here to make things simpler:

``` bash
$ mv contigs-fixed.fa contigs.fa
```


## Creating an anvi'o contigs database

An anvi'o contigs database will keep all the information related to your contigs: positions of open reading frames, k-mer frequencies for each contigs, where splits start and end, functional and taxonomic annotation of genes, etc. The contigs database is an essential component of everything related to anvi'o metagenomic workflow.

### anvi-gen-contigs-database

The following is the simplest way of creating a contigs database:

``` bash
$ anvi-gen-contigs-database -f contigs.fa -o contigs.db -n 'An example contigs datbase'
```

When you run this command, `anvi-gen-contigs-database` will,

* **Compute k-mer frequencies** for each contig (the default is `4`, but you can change it using `--kmer-size` parameter if you feel adventurous).

* **Soft-split contigs** longer than 20,000 bp into smaller ones (you can change the split size using the `--split-length`). When the gene calling step is not skipped, the process of splitting contigs will consider where genes are and avoid cutting genes in the middle. For very very large assemblies this process can take a while, and you can skip it with `--skip-mindful-splitting` flag.

* **Identify open reading frames** using [Prodigal](http://prodigal.ornl.gov/), the bacterial and archaeal gene finding program developed at Oak Ridge National Laboratory and the University of Tennessee. If you don't want gene calling to be done, you can use the flag `--skip-gene-calling` to skip it. If you have your *own* gene calls, you can provide them to be used to identify where genes are in your contigs. All you need to do is to use the parameter `--external-gene-calls` (see down below for the format).

Almost every anvi'o program comes with a help menu that explains available parameters in detail. Don't forget to check them once in a while. If something is not clearly explained, please let us know so we can fix that:

``` bash
$ anvi-gen-contigs-database --help
```

Once you have your contigs database, you can start importing things into it, or directly go to the profiling step.

---

<b><a name="external-gene-calls" style="color: black;">The</a> TAB-delimited file for `--external-gene-calls`</b> should follow this format:

|gene_callers_id|contig|start|stop|direction|partial|source|version|
|:---:|:---|:---:|:---:|:---:|:---:|:---:|
|1|contig_01|1113|1677|f|0|program|v1.0|
|2|contig_01|1698|2142|f|0|program|v1.0|
|3|contig_01|2229|3447|f|0|program|v1.0|
|4|contig_01|3439|6820|r|0|program|v1.0|
|7|contig_01|8496|10350|r|1|program|v1.0|
|8|contig_02|306|1650|f|0|program|v1.0|
|9|contig_02|1971|3132|f|0|program|v1.0|
|10|contig_02|3230|4007|f|0|program|v1.0|
|11|contig_02|4080|5202|f|0|program|v1.0|
|12|contig_02|5194|5926|f|0|program|v1.0|
|13|contig_03|606|2514|f|0|program|v1.0|
|14|contig_03|2751|3207|f|0|program|v1.0|
|15|contig_03|3219|5616|f|0|program|v1.0|
|16|contig_03|5720|6233|f|0|program|v1.0|
|(...)|(...)|(...)|(...)|(...)|(...)|(...)|


{:.notice}
**Please note**: For a complete gene call where the value of `partial` is `0`, the number of nucleotides, i.e. the value of `stop - start`, must be divided by three **without** any remainder. Any other gene call is not a complete gene call, and must be identified as such by setting the value of `partial` to `1` so anvi'o does not try to translate their sequence. Otherwise anvi'o will be upset, and then frustrate you.

{:.notice}
**Please note**: anvi'o follows the convention of string indexing and splicing that is identical to the way one does it in Python or C.

The statement above means that the index of the first nucleotide in any contig should be `0`. In other words, for a gene call that starts at the position `x`th position and ends at position `y`th position, we start counting from `x-1`, and not from `x` (but we still end at `y`. The `start` and `stop` positions in the input file should comply with this. Here is an example gene in a contig:

``` bash
                 1         2         3
nt pos: 12345678901234567890123456789012 (...)
contig: NNNATGNNNNNNNNNNNNNNNNNTAGAAAAAA (...)
           |______ gene X _______|
```

The `start` and `stop` positions in the input file for this gene should be `3` and `26`, respectively. Which means, if you are trying to generate an external gene calls file from gene calls produced by a gene caller that reports start/stop positions starting with the index of `1` rather than `0`, you basically need to substract one from the start position of every gene call for a matching anvi'o external gene calls file.

Gene `start` and `stop` positions do not care about the direction of the gene as they simply address how the gene sequence should be sliced out from a longer sequence. Whether a gene is forward or reverse is defined in the column `direction`.

{:.notice}
You can read the previous discussions regarding this behavior in [this issue](https://github.com/meren/anvio/issues/374)). Thanks for your patience!

### anvi-run-hmms

Although this is absolutely optional, you shouldn't skip this step. Anvi'o can do a lot with hidden Markov models ([HMMs](https://en.wikipedia.org/wiki/Hidden_Markov_model) provide statistical means to model complex data in probabilistic terms that can be used to search for patterns, which works beautifully in bioinformatics where we create models from known sequences, and then search for those patterns rapidly in a pool of unknown sequences to recover hits). To decorate your contigs database with hits from HMM models that ship with the platform (which, at this point, constitute multiple published bacterial single-copy gene collections), run this command:

``` bash
$ anvi-run-hmms -c contigs.db
```

When you run this command (without any other parameters),

* It will utilize multiple default bacterial single-copy core gene collections and identify hits among your genes to those collections using HMMER. If you have already run this once, and now would like to add an HMM profile of your own, that is easy. You can use `--hmm-profile-dir` parameter to declare where should anvi'o look for it. Or you can use the `--installed-hmm-profile` parameter to only run a specific default HMM profile on your contigs database.

* Note that the program will use only one CPU by default, especially if you have multiple of them available to you, you should use the `--num-threads` parameter. It significantly improves the runtime, since [HMMER](http://hmmer.org/) is truly an awesome software.

### anvi-display-contigs-stats

Once you have your contigs database ready, and optionally your HMMs are run, you can take a quick look at it using the program `anvi-display-contigs-stats` program. 

``` bash
$ anvi-display-contigs-stats contigs.db
```

This program shows you simple stats of your contigs database that may help you not only assess your assembly output, but also estimate the number of bacterial and archaeal genomes to recover.

{:.notice}
This program accepts multiple anvi'o contigs databases to compare to each other. You can find some screenshots of how it looks like in [release notes for anvi'o `v3`](https://github.com/merenlab/anvio/releases/tag/v3).

### anvi-run-ncbi-cogs

Yet another optional step is to run the program `anvi-run-ncbi-cogs` to annotate genes in your contigs database with functions from the NCBI's [Clusters of Orthologus Groups](https://www.ncbi.nlm.nih.gov/COG/). You will be glad that you did it later!

Do not forget to use `--num-threads` to specify how many cores you wish to use for that. See the help menu for other available options.

{:.notice}
If you are running COGs for the first time, you will need to set them up on your computer using `anvi-setup-ncbi-cogs`. Simply run it on a machine with an internet connection, and let it do its magic.

### anvi-import-functions

Anvi'o also can make good use of functional annotations you already have for your genes. The following post describes multiple ways to import functions into anvi'o:

- [**Importing functions into anvi'o**]({% post_url anvio/2016-06-18-importing-functions %}).


### anvi-import-taxonomy

Annotating genes with taxonomy can make things downstream more meaningful, and in some cases may improve the human guided binning and refinement steps. Please see this post to find out different ways to achieve that:

- [**Importing taxonomy into anvi'o**]({% post_url anvio/2016-06-18-importing-taxonomy %}).

However, gene-level taxonomy is not reliable for making sense of the taxonomy of the resulting metagenome-assembled genomes.

## Profiling BAM files

If you are here, you must be done with your contigs database, and have your BAM files ready. Good! It is time to initialize your BAM file, and create an *anvi'o profile* for your sample.

### anvi-init-bam

Anvi'o requires BAM files to be sorted and indexed. In most cases the BAM file you get back from your mapping software will not be sorted and indexed. This is why we named the BAM file for our mock samples as `SAMPLE-01-RAW.bam`, instead of `SAMPLE-01.bam`.

If your BAM files already sorted and indexed (i.e., for each `.bam` file you have, there also is a `.bam.bai` file in the same directory), you can skip this step. Otherwise, you need to initialize your BAM files:

``` bash
$ anvi-init-bam SAMPLE-01-RAW.bam -o SAMPLE-01.bam
```

But of course it is not fun to do every BAM file you have one by one. So what to do?

A slightly better way to do would require you to do it in a `for` loop. First, create a file called, say, `SAMPLE_IDs`. For your samples (`X` and `Y`) it will look like this:

``` bash
$ cat SAMPLE_IDs
SAMPLE-01
SAMPLE-02
SAMPLE-03
```

Then, you can run `anvi-init-bam` on all of them by typing this:

``` bash
for sample in `cat SAMPLE_IDs`; do anvi-init-bam $sample-RAW.bam -o $sample.bam; done
```

Of course, if you have a way to cluster your runs, you already know what to do.

One last note, `anvi-init-bam` uses [samtools](https://github.com/samtools/samtools) in the background to do sorting and indexing. While clearly a big thanks and all the credit go to samtools developers, this is also a reminder that you can get your BAM files sorted and indexed using the samtools command line client without using `anvi-init-bam`.

OK. Good.

### anvi-profile

In contrast to the contigs database, an anvi'o profile database stores sample-specific information about contigs. Profiling a BAM file with anvi'o using `anvi-profile` creates a single profile that reports properties for each contig in a single sample based on mapping results. Each profile database links to a contigs database, and anvi’o can merge single profiles that link to the same contigs database into merged profiles (which will be covered later).

In other words, the profiling step makes sense of each BAM file separately by utilizing the information stored in the contigs database. It is one of the most critical (and also most complex and computationally demanding) steps of the metagenomic workflow.

The simplest form of the command that starts the profiling looks like this:

``` bash
$ anvi-profile -i SAMPLE-01.bam -c contigs.db
```

When you run `anvi-profile` it will,

* Process each contig that is longer than `2,500 nts` by default. You can change this value by using the `--min-contig-length` flag. But you should remember that the minimum contig length should be long enough for tetra-nucleotide frequencies to have enough meaningful signal. There is no way to define a golden number for minimum length that would be applicable to genomes found in all environments. We empirically chose the default to be 2,500, and have been happy with it. You are welcome to experiment, but we advise you to never go below 1,000. You also should remember that the lower you go, the more time it will take to analyze all contigs. You can use the --list-contigs parameter to have an idea how many contigs would be discarded for a given `--min-contig-length` parameter. If you have an arbitrary list of contigs you want to profile, you can use the flag `--contigs-of-interest` to ignore the rest.

* Make up some crappy output directory, and sample names for you. We encourage you to use the `--output-dir` parameter to tell anvi'o where to store your output files, and the `--sample-name` parameter to give a meaningful, preferably not-so-long sample name to be stored in the profile database. This name will appear almost everywhere, and changing it later will be a pain.

Processing of contigs will include,

* The recovery of mean coverage, standard deviation of coverage, and the average coverage for the inner quartiles (Q1 and Q3) for a given contig. Profiling will also create an HD5 file where the coverage value for *each nucleotide position* will be kept for each contig for later use. While the profiling recovers all the coverage information, it can discard some contigs with very low coverage declared by `--min-mean-coverage` parameter (the default is 0, so everything is kept).

* The characterization of single-nucleotide variants (SNVs) for every nucleotide position, unless you use `--skip-SNV-profiling` flag to skip it altogether (you will definitely gain a lot of time if you do that, but then, you know, maybe you shouldn't). By default, the profiler will not pay attention to any nucleotide position with less than `10X` coverage. You can change this behavior via `--min-coverage-for-variability` flag. Anvi'o uses a conservative heuristic to not report every position with variation: i.e., if you have 200X coverage in a position, and only one of the bases disagree with the reference or consensus nucleotide, it is very likely that this is due to a mapping or sequencing error, and anvi'o tries to avoid those positions. If you want anvi'o to report everything, you can use `--report-variability-full` flag. We encourage you to experiment with it, maybe with a small set of contigs, but in general you should refrain reporting everything (it will make your databases grow larger and larger, and everything will take longer for -99% of the time- no good reason). 

* Finally, because single profiles are rarely used for genome binning or visualization, and since the clustering step increases the profiling runtime for no good reason, the default behavior of profiling is to *not cluster* contigs automatically. However, if you are planning to work with single profiles, and if you would like to visualize them using the interactive interface without any merging, you can use the `--cluster-contigs` flag to initiate clustering of contigs. In this case anvi'o would use [default clustering configurations for single profiles](https://github.com/meren/anvio/tree/master/anvio/data/clusterconfigs/single), and store resulting trees in the profile database. You *do not* need to use this flag if you are planning to merge multiple profiles (i.e., if you have more than one BAM file to work with, which will be the case for most people).

{:.notice}
Every anvi'o profile that will be merged later must be generated with the same exact parameters and against the same contigs database. Otherwise, anvi'o will complain about it later, and likely nothing will get merged. Just saying.


## Working with anvi'o profiles

You have all your BAM files profiled! Did it take forever? Well, sorry about that. But now you are golden.

### anvi-merge

The next step in the workflow is to to merge all anvi'o profiles.

This is the simplest form of the `anvi-merge` command:

``` bash
$ anvi-merge SAMPLE-01/PROFILE.db SAMPLE-02/PROFILE.db SAMPLE-03/PROFILE.db -o SAMPLES-MERGED -c contigs.db
```

Or alternatively you can run it like this (if your work directory contains multiple samples to be merged):

``` bash
$ anvi-merge */PROFILE.db -o SAMPLES-MERGED -c contigs.db
```

{:.notice}
Please don't forget to give a short, simple, and descriptive sample name using the `--sample-name` parameter, because it will appear in a lot of places later.

When you run `anvi-merge`,

* It will merge everything and create a merged profile (yes, thanks, captain obvious),

* It will attempt to create multiple clusterings of your splits using the default _clustering configurations_. Please take a quick look at the default [clustering configurations for merged profiles](https://github.com/meren/anvio/tree/master/anvio/data/clusterconfigs/merged) --they are pretty easy to understand. By default, anvi'o will use euclidean distance and ward linkage algorithm to organize contigs; however, you can change those default values with the `--distance` and `--linkage` parameters (available options for distance metrics and linkage algorithms are listed in [this release note](https://github.com/meren/anvio/releases/tag/v2.0.2)). Hierarchical clustering results are necessary for comprehensive visualization and human guided binning; therefore, by default, anvi'o attempts to cluster your contigs using default configurations. You can skip this step by using `--skip-hierarchical-clustering` flag. But even if you don't skip it, anvi'o will skip it for you if you have more than 20,000 splits, since the computational complexity of this process will get less and less feasible with increasing number of splits. That's OK, though. There are many ways to recover from this. On the other hand, if you want to teach everyone who is the boss, you can force anvi'o try to cluster your splits regardless of how many of them are there by using `--enforce-hierarchical-clustering` flag. You have the power.

* It will attempt to run [CONCOCT](http://www.nature.com/nmeth/journal/v11/n11/full/nmeth.3103.html) to bin your splits automatically. CONCOCT can deal with hundreds of thousands of splits. Which means, regardless of the number of splits you have, and even if you skip the hierarchical clustering step, there will be a collection in the merged profile database (which will be called `CONCOCT`) with genome bins identified by CONCOCT in an automatic manner. From these you can generate a summary, or run the interactive interface with `--collection-name CONCOCT` parameter (more later on these). But if you would like to skip default CONCOCT clustering, you can use `--skip-concoct-binning` flag. 


### anvi-import-collection

If you have your own binning of your contigs, you can easily import those results into the merged profile database as a collection:

``` bash
$ anvi-import-collection binning_results.txt -p SAMPLES-MERGED/PROFILE.db -c contigs.db --source "SOURCE_NAME"
```

The file format for `binning_results.txt` is very simple. This is supposed to be a TAB-delimited file that contains information about which contig belongs to what bin. So each line of this TAB-delimited file should contain a contig name (or split name, see below), and the bin name it belongs to. If you would like to see some example files, you can find them [here](https://github.com/meren/anvio/tree/master/anvio/tests/sandbox/example_files_for_external_binning_results). They will help you see the difference between input files for splits and contigs after reading the following bullet points, and demonstrate the structure of the optional "bins information" file.


Two points:

* It is common that we use `anvi-export-splits-and-coverages` to export coverage and sequence composition information to bin our contigs with software that can work with coverage and sequence composition information. In this case, our `binning_results.txt` contains *split names*. But if you have contig names, you can import them using `anvi-import-collection` with the flag `--contigs-mode`. 

* You can also use an information file with the `--bins-info` parameter to describe the source of your bins (and even assign them some colors to have some specific visual identifiers for any type of visualization downstream).

{:.notice}
You can use `anvi-export-collection` to export collection information and import into other profiles. It becomes very handy when you are doing [benchmarking between different approaches](% post_url anvio/2015-06-23-comparing-different-mapping-software %).

{:.notice}
You can use `anvi-show-collections-and-bins` to see all available collections and bins in an anvi'o profile or pan database.

{:.notice}
You can use `anvi-script-get-collection-info` to see completion and redundancy estimates for all bins in a given anvi'o collection.


### anvi-interactive

Anvi'o interactive interface is one of the most sophisticated parts of anvi'o. In the context of the metagenomic workflow, the interactive interface allows you to browse your data in an intuitive way as it shows multiple aspects of your data, visualize the results of unsupervised binning, perform supervised binning, or refine existing bins.

{:.notice}
The interactive interface of anvi'o is written from scratch, and can do much more than what is mentioned above. In fact, you don't even need anvi'o profiles to visualize your data using the interactive interface. But since this is a tutorial for the metagenomic workflow, we will save you from these details. If you are interested in learning more, we have other resources that provide **detailed descriptions of [the anvi'o interactive interface and data formats it works with]({% post_url anvio/2016-02-27-the-anvio-interactive-interface %})**.

{:.notice}
Most things you did so far (creating a contigs database, profiling your BAM files, merging them, etc) required you to work on a server. But `anvi-interactive` will require you to download the merged directory and your contigs databases to your own computer, because `anvi-interactive` uses a browser to interact with you. If you don't want to download anything, you can dig an SSH tunnel to use your server to run `anvi-interactive`, and the browser on your computer to interact with it. See the post on **[visualizing from a server]({% post_url anvio/2015-11-28-visualizing-from-a-server %})**.

This is the simplest way to run the interactive interface on your merged anvi'o profile: 

``` bash
$ anvi-interactive -p SAMPLES-MERGED/PROFILE.db -c contigs.db
```

This will work perfectly **if your merged profile has its own trees** (i.e., the hierarchical clustering mentioned in the `anvi-merge` section was done).

<div class="extra-info" markdown="1">

<span class="extra-info-header">Collection mode</span>

If there are no clusterings available in your profile database `anvi-interactive` will complain about the fact that it can't visualize your profile. But if you have an anvi'o collection stored in your profile database, you can run the interactive interface in **collection mode**. If you are not sure whether you have a collection or not, you can see all available collections using this command:

``` bash
$ anvi-script-get-collection-info -p SAMPLES-MERGED/PROFILE.db -c contigs.db --list-collections
```

Once you know the collection you want to work with, you can use this notation to visualize it:

``` bash
$ anvi-interactive -p SAMPLES-MERGED/PROFILE.db -c contigs.db -C CONCOCT
```

When you run `anvi-interactive` with a collection name, it will compute various characteristics of each bin on-the-fly, i.e., their mean coverage, variability, completion and redundancy estimates, and generate anvi'o views for them to display their distribution across your samples in the interactive interface. Briefly, each *leaf* of the anvi'o central dendrogram will represent a "bin" in your collection, instead of a "contig" in your metagenomic assembly. A dendrogram for bins will be generated for each view using euclidean distance and ward linkage automatically. When running the interactive interface in collection mode, you can change those defaults using the `--distance` and/or `--linkage` parameters. If you have run `anvi-merge` with the `--skip-hierarchical-clustering` parameter due to the large number of contigs you had, but you have binning results available to you from an external resource, you can import those bins as described in the previous section, and run the interactive interface with that collection id to immediately see the distribution of bins across your samples.

{:.notice}
In this mode each leaf of the tree will be a bin, along with the distribution of each bin across samples with their completion and redundancy estimates in the most outer layers. In this mode, the interface runs in reduced functionality, and selections will not have completion and contamination estimates. If you are interested in visualizing a specific bin with, say, high redundancy, then you can use the program `anvi-refine` with that bin.

</div>

Here is some additional information about the interactive interface (please see a full list of other options by typing `anvi-interactive -h`):

- **Storing visual settings**. The interactive interface allows you to tweak your presentation of data remarkably, and store these settings in profile databases as "*states*". If there is a state stored with the name `default`, or if you specify a state when you are running the program via the `--state` parameter, the interactive interface will load it, and proceeds to visualize the data automatically (without waiting for you to click the Draw button). States are simply JSON formatted data, and you can export them from or import them into an anvi'o profile database using the `anvi-export-state` and `anvi-import-state` programs. This way you can programmatically edit state files if necessary, and/or use the same state file for multiple projects.

- **Using additional data files**. When you need to display more than what is stored in anvi'o databases for a project, you can pass additional data files to the interactive interface. If you have a newick tree as an alternative organization of your contigs, you can pass it using the `--tree` parameter, and it would be added to the relevant combo box. If you have extra layers to show around the tree (see [Figure 1 in this publication](https://peerj.com/articles/1839/) as an example), you can use the `--additional-layers` parameter. Similarly, you can pass an extra view using the `--additional-view` parameter. Files for both additional layers and additional view are expected to be TAB-delimited text files and information is to be provided at the split-level (if you hate the way we do this please let us know, and we will be like "*alright alright*" and finally fix it). Please see the help menu for more information about the expected format for these files.

- **Setting a taxonomic level**. When information about taxonomy is available in an anvi'o contigs database, anvi'o interactive interface will start utilizing it automatically and you will see a taxonomy layer in the interface for each of your contigs. By default the genus names will be used, however, you can change that behavior using the `--taxonomic-level` flag.


### anvi-[import|export|show|delete]-misc-data

Anvi'o profile databases allow you to add or remove additional data for your items or layers. This is a very important functionality for better data exploration and communication. Please see [this post]({% post_url anvio/2017-12-11-additional-data-tables %}) for more information and to familiarize yourself with it.

{:.notice}
This functionality will not be available to you if you are using anvi'o `v3` or earlier. Please make sure you are using the latest stable version of anvi'o, which is `v{% include _project-anvio-version-number.html %}`.

### anvi-summarize

A *collection* represents one or more bins with one or more contigs. Collections are stored in anvi'o databases can be imported from the results of external binning software, or saved through the anvi-interactive interface after a human-guided binning effort. Once you have a collection, you can *summarize* it using `anvi-summarize`.

{:.notice}
If you don't know what collections and bins are available in a profile database, you can use the program `anvi-show-collections-and-bins`, and if you would like to get a very quick list of completion estimates for your bins in a collection, you can use the program `anvi-script-get-collection-info`.

The result of `anvi-summarize` is a static HTML output that you can visualize in your browser, send to your colleagues, put on your web page (an example from one of our papers is [here](http://anvio.org/data/INFANT-CLC-SUMMARY-SUPERVISED/)), or attach to your submission as a supplementary data for review. When you run `anvi-summarize`,

* All your splits will be merged back to contigs and stored as FASTA files,

* Completion and redundancy estimates for each bin in a collection will be computed and stored in the output,

* TAB-delimited matrix files will be generated for genome bins across your samples with respect to their mean coverage, variability, etc.

You can run this to summarize the bins saved under the collection id 'CONCOCT' into the SAMPLES-SUMMARY directory:

``` bash
anvi-summarize -p SAMPLES-MERGED/PROFILE.db -c contigs.db -o SAMPLES-SUMMARY -C CONCOCT
```

If you are not sure which collections are available to you, you can always see a list of them by running this command:

``` bash
anvi-summarize -p SAMPLES-MERGED/PROFILE.db -c contigs.db --list-collections
```

The summary process can take a lot of time. If you want to take a quick look to identify which bins need to be refined, you can run `anvi-summarize` with `--quick-summary` flag.


### anvi-refine

After running `anvi-summarize`, you may realize that you are not happy with one or more of your bins. This often is the case when you are working with very large datasets and when you are forced to skip the human guided binning step. `anvi-refine` gives you the ability to make finer adjustments to a bin that may be contaminated.

After you refine bins that need attention, you can re-generate your static summary.

{:.notice}
Speaking of which, please take a look at this post where Meren talks about [assessing completion and contamination of metagenome-assembled genomes]({% post_url miscellaneous/2016-06-09-assessing-completion-and-contamination-of-MAGs %}).

Please read **[this article]({% post_url anvio/2015-05-11-anvi-refine %})** for a comprehensive introduction to the refinement capacity of anvi'o. Plus, there are the following articles from Tom Delmont and Veronika Kivenson that you may consider reading:

* [Notes on genome refinement with anvi'o]({% post_url anvio/2017-05-11-anvi-refine-by-veronika %})
* [Inspecting the genomic link between Archaea and Eukaryota]({% post_url miscellaneous/2017-01-03-loki-the-link-archaea-eukaryota %}) 


---

## FAQ

{% include _join-anvio-slack.html %}

### I ran into an issue, whose fault is it?

It is Tom's. But you can always enter a [bug report](https://github.com/meren/anvio/issues) if you are certain that it needs to be fixed in the anvi'o code base. If you are not sure, please use the button above to join our Slack group, or come to [the anvi'o discussion group]({% post_url anvio/2015-10-14-anvio-discussion-group %}), and let's talk about it!

### There is something in this tutorial I want to fix

{% include _fixthispage.html source="_posts/anvio/2016-06-22-anvio-tutorial-v2.md" %}
