---
layout: post
authors: [matt, Chiara]
title: "Agnostos Workflow in Genomes and Metagenomes with anvi'o"
excerpt: "Explore the unknown!"
modified: 2020-07-17
tags: []
categories: [anvio]
comments: true
redirect_from:
  - /agnostos-tutorial/
image:
    feature:
---

{% include _toc.html %}

How many times have you analyzed a metagenome and found a large proportion of the genes have no annotation?

Have you ever been interested in characterizing a bacterial genomic locus but it contains a critical gene that has no determined function?

**The Agnostos workflow** ([Vanni et al., 2020](https://www.biorxiv.org/content/10.1101/2020.06.30.180448v1)) is a great way to start to address the problems above and utilize all genes, Known and Unknown, from microbiomes!

Recently, Vanni et al., 2020 published the preprint [Light into the darkness: Unifying the known and unknown coding sequence space in microbiome analyses](https://www.biorxiv.org/content/10.1101/2020.06.30.180448v1) showcasing the Agnostos Workflow and describing the unknown sequence landscape of microbial communities from ocean and human body biomes. One of the goals of this manuscript was to provide a resource (workflow and dataset) for the community to explore genes of unknown function in their own microbial genomes and metagenomes. Briefly, the Agnostos workflow clustered open reading frames (ORFS) from 1,749 metagenomes and 28,941 bacterial and archaeal genomes, then systematically validated and categorized the clusters into Known and Unknown gene clusters. If you are more interested in the details of the manuscript please refer to the [preprint](https://www.biorxiv.org/content/10.1101/2020.06.30.180448v1). To accompany this manuscript, my former supervisor [Antonio Fernandez-Guerra](http://orcid.org/0000-0002-8679-490X) wrote a blog post that goes over the origins of the Agnostos worflow, the categories of gene-clusters that result from the workflow, and the future of Unknown genes in microbiomes. I highly recommend reading the post for a casual introduction into utilizing the unknown fraction of microbiomes. Check it out [here](http://merenlab.org/2020/07/01/dark-side/#a-conceptual-framework-to-unify-the-known-and-the-unknown-in-microbiome-analyses).

"Ok great, we have this fancy resource... but how do I use it? I have metagenomes I'm analyzing right now!"

To introduce how to utilize the [Agnostos Workflow](https://github.com/functional-dark-side/agnostos-wf), we have composed a tutorial showing how to integrate the output of the workflow from [Vanni et al., 2020](https://www.biorxiv.org/content/10.1101/2020.06.30.180448v1) into your analyses using anvio!

{:.notice}
If you would like to integrate your own microbial genomes or metagenomes in the Agnostos worflow, please refer to the [github repo](https://github.com/functional-dark-side/agnostos-wf) for instructions.

To have some fun, [Chiara Vanni](https://orcid.org/0000-0002-1124-1147) kindly integrated all of the ORFs from the classic anvio metagenomic sandbox, the Infant Gut Dataset (IGD) from [Sharon et al., 2013](http://www.ncbi.nlm.nih.gov/pubmed/22936250)(IGD). We will explore how the Agnostos categories can be used to get more out of both the IGD metagenomes and a pangenome of *E. faecalis*.

If you want straight to data analysis, please skip ahead to the `Download the goods section`. However, if you are interested in integrating your [contigsDB](https://merenlab.org/software/anvio/help/main/artifacts/contigs-db/)s into agnostos then keep reading. This is how Chiara integrated the IGD dataset into agnostos:

Let's get started!

# anvi'o to agnostos

The IGD can be downloaded following the infant-gut tutorial instructions [here](https://merenlab.org/tutorials/infant-gut/#downloading-the-pre-packaged-infant-gut-dataset). Once we have the anvi'o [contigsDB](https://merenlab.org/software/anvio/help/main/artifacts/contigs-db/) we need to extract the predicted gene sequences the following way:

```bash
anvi-get-sequences-for-gene-calls -c CONTIGS.db \
--get-aa-sequences -o infant_gut_genes.fasta
```

Next, we need the [external-gene-calls](https://merenlab.org/software/anvio/help/main/artifacts/external-gene-calls/) table from the [contigsDB](https://merenlab.org/software/anvio/help/main/artifacts/contigs-db/) which contains gene predictions and their coordinates:

```bash
anvi-export-gene-calls -c CONTIGS.db \
-o infant_gut_gene_calls.tsv
```

Finally, to get this table ready for AGNOSTOS, we need to convert the completeness of each gene a double-digits format. Luckily we can do that with this oneliner:

```bash
awk 'NR>1{if($6==0) print $1"\t""00"; else print $1"\t""11";}' infant_gut_gene_calls.tsv > infant_gut_gene_partial_info.tsv
```

The genes amino acid sequences and the completion information (`infant_gut_genes.fasta`, `infant_gut_gene_partial_info.tsv`) are the only data necessary for the integration in an AGNOSTOS-DB.

To proceed with the integration, specify the paths to these files together with the paths to input and output folders i n the agnostos-wf [configuration file](https://github.com/functional-dark-side/agnostos-wf/blob/master/db_update/config/config.yaml) and follow the rest of the instructions provided in the [workflow GitHub page](https://github.com/functional-dark-side/agnostos-wf).

That was easy right?? Moving on to data analysis.

# Download the goods

First, let's make a directory where we will work with the IGD Agnostos integrated data and download all the files we will need.
```bash
# Download IGD data
wget https://ndownloader.figshare.com/files/18046139 -O INFANT-GUT-TUTORIAL.tar.gz
tar -zxvf INFANT-GUT-TUTORIAL.tar.gz && cd INFANT-GUT-TUTORIAL

# Make sure all the DBs are up to date
anvi-migrate --migrate-dbs-safely *.db

# Make a home for the agnostos data
mkdir -p IGD_agnostos

#### CV
# The summary outputs of the integration are stored in the results folder "output_tables"
# For the pangenomic part of the tutorial, we also integrated into agnostos the predicted genes retrieved from the collection of external genomes used/described [here](https://merenlab.org/tutorials/infant-gut/#chapter-iv-pangenomics)
####

# Download agnostos integrated IGD data
wget https://ndownloader.figshare.com/files/24028721 -O IGD_agnostos/IGD_genes_summary_info_exp.tsv

# Download E. faecalis integrated data
wget https://ndownloader.figshare.com/files/24028730 -O IGD_agnostos/IGD_ext_genomes_summary_info_exp.tsv
```

Check out a description of the columns in the file [here](https://github.com/functional-dark-side/agnostos-wf/blob/master/Output_README.md).

# Metagenomics applications

Now let's get the Unknowns party started!

Metagenomic sequencing allows access to the uncultured majority of microbiomes all kinds of biomes including soil, ocean, to the human gut. These metagenomic samples are a huge source of genetic novelty. When we explore metagenomic samples one of the most insightful steps is annotating all of the assembled genes, but unfortunately, sometimes more than half can come back with no annotation. For those genes, the analysis stops there.

Here we will see how to use Agnostos categories will allow us to utilize the entire set of genes from metagenomes instead of just the annotated ones. To demonstrate this idea we will ask some basic questions from the IGD metagenomes after they have been **integrated** into the Agnostos gene-clusters via the workflow. The concept of ORF integration is key here because we are not passing annotations to the IGD ORFs through alignment based approaches (e.g. BLAST, hmmsearch), but are attempting to insert the IGD ORFs into the extant Agnostos clusters that already have metadata associated with them (read more about it [here](http://merenlab.org/2020/07/01/dark-side/#why-gene-clusters)). Integration allows for the entire dataset of ORFs to be used rather than alignment based methods where one filter for best hits. Additionally, when IGD ORFs do no find a home in an Agnostos cluster, the worflow attempts to form new clusters (we will explore this in a second)!

## Import Agnostos into anvi'o

Let's explore the Agnostos integrated IGD data. The first step will be to import the data as "functions" into the IGD [contigs DB](http://merenlab.org/software/anvio/help/artifacts/contigs-db/) (check out this [post](http://merenlab.org/2016/06/18/importing-functions/) if you have more questions about anvio functions tables). This will make it easy to explore the Agnostos categories in the context of assembled contigs and read recruitment results in the anvi'o interactive interface!

First, let's import that agnostos data into the IGD [contigs DB](http://merenlab.org/software/anvio/help/artifacts/contigs-db/):
```bash
anvi-import-functions -c CONTIGS.db -p agnostos -i IGD_agnostos/IGD_genes_summary_info_exp.tsv
```

## Metagenome assembled genome applications

Now that we have imported the Agnostos categories into our [contigs DB](http://merenlab.org/software/anvio/help/artifacts/contigs-db/), let's leverage them and explore Unknowns in metagenomic-assembled genomes (MAGs)!

Because we already imported the Agnostos output into anvi'o, we can immediately visualize the results like this:

```
anvi-interactive -p PROFILE.db -c CONTIGS.db
```
[![agnostos_binning](/images/agnostos_binning.png)](images/agnostos_binning.png){:.center-img .width-100}

[![agnostos_binning](/images/agnostos_legend.jpeg)](images/agnostos_legend.jpeg){:.center-img .width-2}


# Pangenomic applications

The Agnostos workflow is a great way to get more out of genes of unknown function when comparing genomes. For example, one could could find a set of accessory genes that separate genomes into ecotypes or explore genes of unknown function in genomic islands! [Chiara Vanni](https://orcid.org/0000-0002-1124-1147) also kindly integrated a set of *E. faecalis* external genomes for us to work with. Let's calculate a pangenomes of 6 *E. faecalis* genomes! This will be broken down into a few steps: (1) Import the Agnostos categories to the external as functions, (2) Calculate the pangenome, (3) Extract the gene-cluster information with associated Agnostos data and create a miscellaneous layer to import into anvio, (4) Visualize.


Separate agnostos data into individual genome files.
```r
external_genomes_agnostos_categories <- read_tsv("IGD_agnostos/IGD_ext_genomes_summary_info_exp.tsv")

external_genomes_agnostos_categories %>%
  mutate(genome = gsub('(.*)_\\w+', '\\1', gene)) %>%
  write_tsv("IGD_agnostos/IGD_ext_genomes_summary_info_exp_ed.tsv")


genome_list <- external_genomes_agnostos_categories %>%
  mutate(genome = gsub('(.*)_\\w+', '\\1', gene)) %>%   # make new column for genome
  select(-gene) %>%
  group_split(genome)


write_agnostos_genome <- function(DF) {
  file_name <- DF$genome %>% unique()
  file_name
  path_name <- paste0("additional-files/pangenomics/external-genomes/", file_name, "_agnostos_categories.tsv")
  
  DF %>% write_tsv(path_name)
}

lapply(genome_list, write_agnostos_genome)
```
Now we have separate agnostos files for each external genome.

Import Agnostos functions to external E. faecalis external genomes
```bash
# Focus just on E. faecalis
grep -v "faecium" additional-files/pangenomics/external-genomes.txt > additional-files/pangenomics/external-genomes_efaecalis.txt

external_genomes_path="additional-files/pangenomics/external-genomes"

for GENOME in `cut -f 2 additional-files/pangenomics/external-genomes_efaecalis.txt | tail -n +2`;
do
  fname=$(basename $GENOME .db)
  echo "${external_genomes_path}"/"${fname}".db
  echo "${fname}"
  anvi-import-functions -c "${external_genomes_path}"/"${fname}".db -p agnostos -i"${external_genomes_path}"/"${fname}"_agnostos_categories.tsv;
done
```

Calculate the pangenome of E. faecalis:
```bash
# Generate genome storage
anvi-gen-genomes-storage -e additional-files/pangenomics/external-genomes_efaecalis.txt \
                         -o Enterococcus-faecalis-GENOMES.db

# Calculate pangenome
anvi-pan-genome -g Enterococcus-faecalis-GENOMES.db \
                -n Enterococcus_faecalis \
                -o Enterococcus_faecalis \
                --num-threads 14
```


Visualize:
```bash
anvi-display-pan -g Enterococcus-faecalis-GENOMES.db \
                 -p Enterococcus_faecalis/Enterococcus_faecalis-PAN.db \
                 --title "Enterococccus Pan"
```
[![agnostos_pan](/images/agnostos_pan.png)](images/agnostos_pan.png){:.center-img .width-100}

Above we can see the E. faecalis pangenome with an extra layer designating the Agnostos categories. When you focus your attention on the core gene-clusters you can see that most of them are Known. This makes sense because a lot of core, house-keeping genes have been functionally validated. But when you look at the accessory aspects of the pangenome you can see a lot more of the Unknown categories.

[![agnostos_pan_syn](/images/agnostos_pan_syn.png)](images/agnostos_pan_syn.png){:.center-img .width-100}

In the interactive interface you can also re-order the gene-clusters to the synteny of a specific genome, in this case *E. faecalis* 6240. Highlighted in the red boxes, you can see loci that are unique to a select few of the genomes and are enriched in Unknowns.

[![agnostos_ordering_pan](/images/agnostos_ordering_pan.png)](images/agnostos_ordering_pan.png){:.center-img .width-100}

Additionally, you can re-order the gene-clusters by Agnostos category! This is a quick way to find Genomic Unknowns that are core to the pangenome.


# Exploring Contigs

Now that we have a general idea of the distribution of Agnostos categories in IGD, we can leverage genomic context of assembled contigs to possibly infer functions of unknown genes by exploring their genomic context. To do this, we will filter for interesting contigs that have interesting proportions of agnostos categories. To demonstrate, this check out the commands below:

Now that we have an anvio friendly functions table, we can import it into the IGD [contigs DB](http://merenlab.org/software/anvio/help/artifacts/contigs-db/). Then, the best part, we can export the data in the context of metagenomic contigs! This is a powerful method because we can look for contigs that are enriched in Genomic Unknowns and leverage the genomic neighborhood to infer functionality.

Import Agnostos categories into anvio, then summarize:
```bash
# Import functions into metagenome
#anvi-import-functions -c CONTIGS.db -i IGD_functions.tsv
#anvi-import-functions -c CONTIGS.db -i IGD_sources.tsv

# Make a collection of all contigs called EVERYTHING
anvi-script-add-default-collection -p PROFILE.db \
                                   -c CONTIGS.db \
                                   -C EVERYTHING

# Retrieve a ton of awesome tables for fun later
anvi-summarize -c CONTIGS.db \
               -p PROFILE.db \
               -C EVERYTHING \
               --report-aa-seqs-for-gene-calls \
               --init-gene-coverages \
               -o SUMMARY-EVERYTHING
```
Get a list of contigs with intersting proportions of agnostos categories:
```r
contigs_with_agnostos_categories_prop <- all_gene_calls_aggregated %>%
  select(gene_callers_id, contig, start, stop, direction, agnostos_categories_aggregated, COG_FUNCTION) %>%
  group_by(contig) %>%
  mutate(contig_length = length(contig)) %>%
  group_by(contig, agnostos_categories_aggregated) %>%
  mutate(categ_sum = n()) %>%
  mutate(proportion_category = categ_sum/contig_length) %>%
  ungroup() %>%
  select(contig, contig_length, agnostos_categories_aggregated, proportion_category) %>%
  unique() %>%
  spread(agnostos_categories_aggregated, proportion_category, fill = 0)

# Contigs with that we can visualize with Knowns and a few Unknowns
contig_list <- contigs_with_agnostos_categories_prop %>%
  filter(contig_length > 8 & contig_length < 15) %>%
  filter(unknown > 0) %>%
  filter(DISC < 0.1) %>%
  .$contig
```

Visualize potentially interesting contigs using [genoplotR](https://genoplotr.r-forge.r-project.org/):
```r
# Prepare contig data to visualize using genoplotR
contig_vis_info <- all_gene_calls %>%
  mutate(fill = case_when( grepl("GU", agnostos_categories)  ~ 'blue',
                           grepl("EU", agnostos_categories) ~ 'red',
                           grepl("K", agnostos_categories) ~ 'black',
                           grepl("SINGL", agnostos_categories) ~ 'white',
                           grepl("DISC", agnostos_categories) ~ 'white')) %>%
  select(contig, "COG_FUNCTION", start, stop, direction, fill) %>%
  rename(name = "COG_FUNCTION", strand = direction, end = stop) %>%
  mutate(col = "black",
         lty='1',
         lwd='1',
         pch='8',
         cex='1',
         gene_type='arrows') %>%
  mutate(strand = gsub("f", "1", strand)) %>%
  mutate(strand = gsub("r", "-1", strand)) %>%
  mutate(lwd = as.numeric(lwd)) %>%
  mutate(lty = as.numeric(lty)) %>%
  mutate(pch = as.numeric(pch)) %>%
  mutate(cex = as.numeric(cex))

library(genoPlotR)

# function for gene annotation
prep_gene_plot <- function(x){
  mid_pos <- middle(x)             # get position of where you want to put the annotation text over the gene
  annot1 <- annotation(x1=mid_pos, # get names of the genes
                       text=x$name,
                       rot=30)     # rotate the text
  return(annot1)
}

# function for visualizing
visualize_contig <- function(X, contig_list){
  dna1 <- contig_vis_info %>%
    filter(contig %in% contig_list[X]) %>%
    select(-contig)

  dna2 <- contig_vis_info %>%
    filter(contig %in% contig_list[X]) %>%
    select(-contig)

  dna1_seg <- dna_seg(dna1)
  dna2_seg <- dna_seg(dna2)

  dna_list <-  list(dna1_seg,dna2_seg)
  anno_list <- lapply(dna_list[1:2], FUN = prep_gene_plot)

  plot_gene_map(dna_segs=dna_list,annotations = anno_list)
}

# Visualize contig
visualize_contig(X = 5, contig_list)
```

Here is an contig with a GU located right in the middle of a potential secretion operon!

[![agnostos_cat_2](/images/agnostos_contig.png)](images/agnostos_contig.png){:.center-img .width-100}

# Conclusion

The Agnostos workflow from [Vanni et al., 2020](https://www.biorxiv.org/content/10.1101/2020.06.30.180448v1) provides a path forward to utilzing all genes from microbiomes and integerates seamlessly into your standard metagenomic or pangenomic workflow as seen above using anvio. We hope this brief introduction will catalyze your future microbiomes analyses!

Please refer to our recent preprint [Light into the darkness: Unifying the known and unknown coding sequence space in microbiome analyses](https://www.biorxiv.org/content/10.1101/2020.06.30.180448v1) and [blog post](http://merenlab.org/2020/07/01/dark-side/#a-conceptual-framework-to-unify-the-known-and-the-unknown-in-microbiome-analyses) by [Antonio](http://orcid.org/0000-0002-8679-490X) for background and technical details on how to illuminate the genes of unknown function.

<!---- CV
## The following sentence is from Antonio! :)
--->
"To help you stop having to sweep the unknown under the carpet, we are simply removing the carpet." - Antonio



# Orphan code
To do this, we will calculate the proportions of Agnostos categories per bin and visualize it (in IGD). (In the near future this will be a plug-and-play tool in anvio called **anvi-show-agnostos-in-bins**)

First, grab all the bin information from the Infant Gut Tutorial:
```bash
# Important the pre-binned data, thanks Meren!
anvi-import-collection additional-files/collections/merens.txt \
                       --bins-info additional-files/collections/merens-info.txt \
                       -p PROFILE.db \
                       -c CONTIGS.db \
                       -C default

# Export tables with all the info we need
anvi-summarize -p PROFILE.db \
               -c CONTIGS.db \
               -C default \
               -o SUMMARY-BINS
```

Now import Summaries per bin into R:
```r
# construct path string
path <- paste0(getwd(), "/SUMMARY-BINS/bin_by_bin/")

# list the files we need
files <- list.files(path)

# Function to import files
load_bins <- function(files){

  name <- files
  name2 <- paste0(name,  "-gene_calls.txt")

  path_to_gene_calls <- paste0(path, name, "/", name2)

  df <- read_tsv(path_to_gene_calls)

  df  %>%
  mutate(k_unk_category = case_when( grepl("GU", agnostos_categories)  ~ 'UNKNOWN',
                                     grepl("EU", agnostos_categories) ~ 'KNOWN',
                                     grepl("K", agnostos_categories) ~ 'KNOWN',
                                     grepl("SINGL", agnostos_categories) ~ 'SINGL',
                                     grepl("DISC", agnostos_categories) ~ 'DISC',
                                     is.na(agnostos_categories) ~ 'non_integrated')) %>%
  count(k_unk_category) %>%
  mutate(prop = prop.table(n)) %>%
  mutate(bin = name)
}

# Import list of files
bins <- lapply(files[1:13], load_bins)

# Combine into one dataframe
bins_df <- bind_rows(bins)
```

Next, make a miscellaneous data layer for the IGD metagenomes where you show proportions of Agnostos categories per bin in the interactive interface. If you have any questions about adding layers to the anvio interactive interface check out his post [here](http://merenlab.org/2017/12/11/additional-data-tables/)

Create miscellaneous data layer:
```r
bin_agnostos_proportions <- bins_df %>%
  group_by(bin) %>%
  mutate(prop = n/sum(n)) %>%
  select(-n) %>%
  spread( k_unk_category, prop, fill = 0) %>%
  rename("Agnostos_categories!DISC" = DISC,
         "Agnostos_categories!SINGL" = SINGL,
         "Agnostos_categories!KNOWN" = KNOWN,
         "Agnostos_categories!UNKNOWN" = UNKNOWN,
         "Agnostos_categories!non_integrated" = non_integrated)
```

Export table that shows which splits belong to which bin:
```bash
anvi-export-table PROFILE.db --table collections_of_splits -o collection_of_splits.tsv
```

Import split data into R and join with the Agnostos category proportions:
```r
collection_of_splits <- read_tsv("collection_of_splits.tsv")

split_additional_data <- collection_of_splits %>%
  filter(collection_name == "default") %>%
  left_join(bin_agnostos_proportions %>% rename(bin_name = bin)) %>%
  select(split, `Agnostos_categories!DISC`, `Agnostos_categories!SINGL`, `Agnostos_categories!KNOWN`, `Agnostos_categories!non_integrated`, `Agnostos_categories!UNKNOWN`) %>%
  rename(contig = split) %>%
  drop_na()

split_additional_data %>%
  write_tsv("split_additional_data.tsv")
```
Finally, import the miscellaneous data to the IGD [profile DB](http://merenlab.org/software/anvio/help/artifacts/profile-db/) and visualize.
```bash
anvi-import-misc-data split_additional_data.tsv \
                      -p PROFILE.db \
                      -t items
```

```r
# Get Chiara's integrated external genome data
external_genomes_agnostos_categories <- read_tsv("IGD_agnostos/IGD_ext_genomes_summary_info_exp.tsv")

# separate the ORF data by genome
genome_list <- external_genomes_agnostos_categories %>%
  mutate(genome = gsub('(.*)_\\w+', '\\1', gene)) %>%   # make new column for genome
  select(-gene) %>%
  group_split(genome)

# convert data into anvio friendly function import format
export_functions <- function(DF) {
  file_name <- DF$genome %>% unique()

  path_name <- paste0("additional-files/pangenomics/external-genomes/", file_name, "_agnostos_categories.tsv")

  DF %>%
    mutate("source" = "Agnostos_categories") %>%
    select(-genome) %>%
    rename(accession = cl_name, "function" = category) %>%
    mutate("e_value" = 0) %>%
    select(gene_callers_id, source, accession, "function", e_value) %>%
    write_tsv(path_name, col_names = TRUE)
}

# Run it
lapply(genome_list[1:11], export_functions)
```

Now we will import Agnostos categories as (functions) to each contigs DB for the external genomes:
```bash
for GENOME in `ls additional-files/pangenomics/external-genomes/*.db`;
do
  anvi-migrate "${GENOME}" --migrate-dbs-safely # because we've probably updated either contigs or profile dbs
  fname=$(basename "${GENOME}" .db)
  anvi-import-functions -c  "${GENOME}" -i additional-files/pangenomics/external-genomes/"${fname}"_agnostos_categories.tsv;
done
```

Now that we have finished with the external genomes, let's prepare the E. faecalis bin Meren retrieved from the IGD metagenome. To do this, we will separate the E. faecalis bin from the IGD metagenome and import Agnostos categories to it.

First, let's get the E. Faecalis bin from the IGD dataset. Meren already did the binning for you :)

```bash
anvi-import-collection additional-files/collections/e-faecalis.txt \
                       --bins-info additional-files/collections/e-faecalis-info.txt \
                       -p PROFILE.db \
                       -c CONTIGS.db \
                       -C E_faecalis
```

Now we will enter Gene-mode for E. faecalis to get the specific gene-callers-ids with Agnostos categories:
```bash
anvi-script-add-default-collection -p PROFILE.db

anvi-interactive -p PROFILE.db \
                 -c CONTIGS.db \
                 -C default \
                 -b E_facealis \
                 --gene-mode

anvi-export-table GENES/default-E_facealis.db \
                  --table gene_level_coverage_stats \
                  -o GENES/default-E_facealis.tsv
```

## Integrating Agnostos categories into E. faecalis pangenome

Now that we have the we have the Agnostos data integrated into the genomes we can use anvio to make a pangenome!




Now that we have calculated a E. faecalis pangenome, let's extract the Agnostos category information associated with each gene-cluster!

Summarize gene-cluster info:
```bash
# Add default script to pangenome
anvi-script-add-default-collection -p Enterococcus_faecalis/Enterococcus_faecalis-PAN.db

# Get all the awesome tables
anvi-summarize -g Enterococcus-faecalis-GENOMES.db -p Enterococcus_faecalis/Enterococcus_faecalis-PAN.db -C DEFAULT -o SUMMARY-E_faecalis

# unzip the goods
gunzip SUMMARY-E_faecalis/Enterococcus_faecalis_gene_clusters_summary.txt.gz
```

Combine agnostos annotations with gene-clusters:
```r
# Get gene cluster info
Enterococcus_gene_clusters <- read_tsv("SUMMARY-E_faecalis/Enterococcus_faecalis_gene_clusters_summary.txt") %>%
  mutate(genome_name = gsub("E", "Enterococcus", genome_name)) %>%
  rename(gene_caller_id = "gene_callers_id")

# Join info for external gene calls
external_genomes_integrated_agnostos <- read_tsv("IGD_agnostos/IGD_ext_genomes_summary_info_exp.tsv")

# add proportions to external genomes
gene_cluster_additional_data <- Enterococcus_gene_clusters %>%
  unite("gene", c("genome_name", "gene_caller_id"), remove = FALSE, sep = "_") %>%
  left_join(external_genomes_integrated_agnostos) %>%
  select(gene_cluster_id, gene_caller_id, category, genome_name) %>%
  filter(!grepl("SHA", genome_name)) # just get the external genomes

# add proportions to internal genome (MAG)
internal_genome_annotations <- Enterococcus_gene_clusters %>%
  filter(grepl("SHA", genome_name)) %>%
  left_join(IGD_genes_summary_info_exp %>% rename(gene_caller_id = gene_callers_id, category = `function`)) %>%
  select(gene_cluster_id, gene_caller_id, category, genome_name)

gene_cluster_agnostos_categories <- gene_cluster_additional_data %>%
  bind_rows(internal_genome_annotations) %>%
  group_by(gene_cluster_id) %>%
  select(gene_cluster_id, gene_caller_id, category) %>%
  count(category) %>%
  mutate(prop = n/sum(n)) %>%
  select(-n) %>%
  spread(category, prop, fill = 0) %>%
  rename("Agnostos_catergories!DISC" = DISC,
         "Agnostos_catergories!SINGL" = SINGL,
         "Agnostos_catergories!GU" = GU,
         "Agnostos_catergories!K" = K,
         "Agnostos_catergories!KWP" = KWP)

gene_cluster_agnostos_categories %>%
  write_tsv(path = "Enterococcus_faecalis/gene_cluster_additional_data.tsv")
```



