---
layout: post
authors: [matt, chiara]
title: "Integrating AGNOSTOS gene categories into anvi'o projects"
excerpt: "Explore the unknown!"
modified: 2021-05-28
tags: []
categories: [anvio]
comments: true
redirect_from:
  - /agnostos-tutorial/
image:
    feature:
---

{:.warning}
This tutorial is a work-in-progress. We wish to improve it as soon as we can and please feel free to reach out to us if you have any questions.

{% include _toc.html %}

How many times have you analyzed a metagenome and found a large proportion of the genes have no annotation?

Have you ever been interested in characterizing a bacterial genomic locus but it contains a critical gene that has no determined function?

**The Agnostos workflow** ([Vanni et al., 2020](https://www.biorxiv.org/content/10.1101/2020.06.30.180448v1)) is a great way to start to address the problems above and utilize all genes, Known and Unknown, from microbiomes!

Recently, Vanni et al., 2020 published the preprint [Light into the darkness: Unifying the known and unknown coding sequence space in microbiome analyses](https://www.biorxiv.org/content/10.1101/2020.06.30.180448v1) showcasing the Agnostos Workflow and describing the unknown sequence landscape of microbial communities from ocean and human body biomes. One of the goals of this manuscript was to provide a resource (workflow and dataset) for the community to explore genes of unknown function in their own microbial genomes and metagenomes. Briefly, the Agnostos workflow clustered open reading frames (ORFS) from 1,749 metagenomes and 28,941 bacterial and archaeal genomes, then systematically validated and categorized the clusters into Known and Unknown gene clusters. If you are more interested in the details of the manuscript please refer to the [preprint](https://www.biorxiv.org/content/10.1101/2020.06.30.180448v1). To accompany this manuscript, my former supervisor [Antonio Fernandez-Guerra](http://orcid.org/0000-0002-8679-490X) wrote a blog post that goes over the origins of the Agnostos worflow, the categories of gene-clusters that result from the workflow, and the future of Unknown genes in microbiomes. I highly recommend reading the post for a casual introduction into utilizing the unknown fraction of microbiomes. Check it out [here](http://merenlab.org/2020/07/01/dark-side/#a-conceptual-framework-to-unify-the-known-and-the-unknown-in-microbiome-analyses).

"Ok great, we have this fancy resource... but how do I use it? I have metagenomes I'm analyzing right now!"

To introduce how to utilize the [Agnostos Workflow](https://github.com/functional-dark-side/agnostos-wf), we have composed a tutorial showing how to integrate the output of the workflow from [Vanni et al., 2020](https://www.biorxiv.org/content/10.1101/2020.06.30.180448v1) into your analyses using anvio!

{:.notice}
If you would like to integrate your own microbial genomes or metagenomes in the Agnostos worflow, please refer to the [github repo](https://github.com/functional-dark-side/agnostos-wf) for instructions.

To have some fun, [Chiara Vanni](https://orcid.org/0000-0002-1124-1147) kindly integrated all of the ORFs from the classic anvio metagenomic sandbox, the Infant Gut Dataset (IGD) from [Sharon et al., 2013](http://www.ncbi.nlm.nih.gov/pubmed/22936250)(IGD). We will explore how the Agnostos categories can be used to get more out of both the IGD metagenomes and a pangenome of *E. faecalis*.

If you want straight to data analysis, please skip ahead to the `Download the goods section`. However, if you are interested in integrating your {% include ARTIFACT name="contigs-db" text="contigs database" %} into agnostos then keep reading. This is how Chiara integrated the IGD dataset into agnostos:

Let's get started!

## Setup AGNOSTOS on your computer

{:.notice}
This step is only necessary if you want to run AGNOSTOS on your own data. But if your purpose is to simply see how AGNOSTOS output is integrated into an anvi'o project, you can skip this step and use our pre-computed output files generated by AGNOSTOS. 

To have a working AGNOSTOS installation on your computer (or on the computer system you're working), please follow the the [instructions on AGNOSTOS GitHub](https://github.com/functional-dark-side/agnostos-wf) and come back here once you have confirmed that your system is ready!

## Exporting data from anvi'o

The purpose of this section is to give you a demonstration on how to import AGNOSTOS results into an anvi'o project. Here we will use [the Infant Gut Dataset](https://merenlab.org/tutorials/infant-gut/) (IGD) as an example project, but you should be able to follow these instructions to get the necessary information from an anvi'o project to annotate your genes with AGNOSTOS categoreis.

To have a contigs database in our work directory, first follow the download instructions in the IGD tutorial  [here](https://merenlab.org/tutorials/infant-gut/#downloading-the-pre-packaged-infant-gut-dataset) and come back here. Essentially, you should be able to see an output like this in your work directory to follow the rest of the tutorial:

```
AUXILIARY-DATA.db  CONTIGS.db  PROFILE.db  additional-files/
```

Now we can export the gene sequences from the contigs database using the anvi'o program {% include PROGRAM name="anvi-get-sequences-for-gene-calls" %}:

```bash
anvi-get-sequences-for-gene-calls -c CONTIGS.db \
                                  --get-aa-sequences \
                                  -o infant_gut_genes.fasta
```

Next, we need the {% include ARTIFACT name="external-gene-calls" %} information from our contigs database, which contains gene predictions and their coordinates, and can be recovered the following way:

```bash
anvi-export-gene-calls -c CONTIGS.db \
                       --gene-caller prodigal \
                       -o infant_gut_gene_calls.tsv
```

Finally, to get this table ready for AGNOSTOS, we need to create a special file that describes which gene calls were partial in the contigs database. The following one-liner will preapred this input file from the gene calls file produced by anvi'o:

```bash
awk 'NR>1{if($6==0) print $1"\t""00"; else print $1"\t""11";}' \
        infant_gut_gene_calls.tsv > infant_gut_gene_partial_info.tsv
```

At this point we have all the necessary files, namely the gene amino acid sequences (`infant_gut_genes.fasta`) and the gene completion information (`infant_gut_gene_partial_info.tsv`), to recover gene categories from AGNOSTOS-DB.


## Running AGNOSTOS

To proceed with running the AGNOSTOS workflow on your gene sequences, you will need to edit the follwoing variables in the [configuration file](https://github.com/functional-dark-side/agnostos-wf/blob/master/db_update/config/config.yaml) to provide proper paths that point these files:

* Variable 1
* Variable 2
* Variable 3
* (...)

Then run it like this:

```
a one liner from the tutorial?
```

At the end of this, you should expect to have the following files:

* X
* Y

If you are here to follow the tutorial, you can download pre-computed files for the IGD the following way:

```bash

# Make a home for the agnostos data
mkdir -p IGD_agnostos

#### CV
# The summary outputs of the integration are stored in the results folder "output_tables"
# For the pangenomic part of the tutorial, we also integrated into agnostos the predicted genes retrieved from the collection of external genomes used/described [here](https://merenlab.org/tutorials/infant-gut/#chapter-iv-pangenomics)
####

# Download agnostos integrated IGD data
wget https://ndownloader.figshare.com/files/24028721 -O IGD_agnostos/IGD_genes_summary_info_exp.tsv

# Download E. faecalis integrated data
wget https://ndownloader.figshare.com/files/24028730 -O IGD_agnostos/IGD_ext_genomes_summary_info_exp.tsv
```

{:.notice}
Check out a description of the columns in the file [here](https://github.com/functional-dark-side/agnostos-wf/blob/master/Output_README.md).

Now you are ready to import the AGNOSTOS output for your genes back into your anvi'o project.

## Metagenomics applications

Now let's get the Unknowns party started!

Metagenomic sequencing allows access to the uncultured majority of microbiomes all kinds of biomes including soil, ocean, to the human gut. These metagenomic samples are a huge source of genetic novelty. When we explore metagenomic samples one of the most insightful steps is annotating all of the assembled genes, but unfortunately, sometimes more than half can come back with no annotation. For those genes, the analysis stops there.

Here we will see how to use Agnostos categories will allow us to utilize the entire set of genes from metagenomes instead of just the annotated ones. To demonstrate this idea we will ask some basic questions from the IGD metagenomes after they have been **integrated** into the Agnostos gene-clusters via the workflow. The concept of ORF integration is key here because we are not passing annotations to the IGD ORFs through alignment based approaches (e.g. BLAST, hmmsearch), but are attempting to insert the IGD ORFs into the extant Agnostos clusters that already have metadata associated with them (read more about it [here](http://merenlab.org/2020/07/01/dark-side/#why-gene-clusters)). Integration allows for the entire dataset of ORFs to be used rather than alignment based methods where one filter for best hits. Additionally, when IGD ORFs do no find a home in an Agnostos cluster, the worflow attempts to form new clusters (we will explore this in a second)!

### Import Agnostos into anvi'o

Let's explore the Agnostos integrated IGD data. The first step will be to import the data as "functions" into the IGD [contigs DB](http://merenlab.org/software/anvio/help/artifacts/contigs-db/) (check out this [post](http://merenlab.org/2016/06/18/importing-functions/) if you have more questions about anvio functions tables). This will make it easy to explore the Agnostos categories in the context of assembled contigs and read recruitment results in the anvi'o interactive interface!

First, let's import that agnostos data into the IGD [contigs DB](http://merenlab.org/software/anvio/help/artifacts/contigs-db/):
```bash
anvi-import-functions -c CONTIGS.db -p agnostos -i IGD_agnostos/IGD_genes_summary_info_exp.tsv
```

### Metagenome assembled genome applications

Now that we have imported the Agnostos categories into our [contigs DB](http://merenlab.org/software/anvio/help/artifacts/contigs-db/), let's leverage them and explore Unknowns in metagenomic-assembled genomes (MAGs)!

First, let's import some bins Meren made:
```
anvi-import-collection additional-files/collections/merens.txt \
                       --bins-info additional-files/collections/merens-info.txt \
                       -p PROFILE.db \
                       -c CONTIGS.db \
                       -C default
```

Now because we already imported the Agnostos output into anvi'o, we can immediately visualize the results like this:
```
anvi-interactive -p PROFILE.db -c CONTIGS.db -F Agnostos
```
[![agnostos_binning](/images/agnostos_binning.png)](images/agnostos_binning.png){:.center-img .width-100}

[![agnostos_binning](/images/agnostos_legend.jpeg)](images/agnostos_legend.jpeg){:.center-img .width-2}


## Pangenomic applications

{:.warning}
We are working on this section! Once we are done, this section will include instructions to add proportion of AGNOSTOS categories for each gene cluster in a given anvi'o pangenome as a layer:

[![agnostos_pan](/images/agnostos_pan.png)](images/agnostos_pan.png){:.center-img .width-100}


# Conclusion

The Agnostos workflow from [Vanni et al., 2020](https://www.biorxiv.org/content/10.1101/2020.06.30.180448v1) provides a path forward to utilzing all genes from microbiomes and integerates seamlessly into your standard metagenomic or pangenomic workflow as seen above using anvio. We hope this brief introduction will catalyze your future microbiomes analyses!

Please refer to our recent preprint [Light into the darkness: Unifying the known and unknown coding sequence space in microbiome analyses](https://www.biorxiv.org/content/10.1101/2020.06.30.180448v1) and [blog post](http://merenlab.org/2020/07/01/dark-side/#a-conceptual-framework-to-unify-the-known-and-the-unknown-in-microbiome-analyses) by [Antonio](http://orcid.org/0000-0002-8679-490X) for background and technical details on how to illuminate the genes of unknown function.

<!---- CV
## The following sentence is from Antonio! :)
--->
"To help you stop having to sweep the unknown under the carpet, we are simply removing the carpet." - Antonio

