---
layout: post
authors: [matt, chiara]
title: "Integrating AGNOSTOS gene categories into anvi'o projects"
excerpt: "Explore the unknown!"
modified: 2021-05-28
tags: []
categories: [anvio]
comments: true
redirect_from:
  - /agnostos-tutorial/
image:
    feature:
---

{:.warning}
This tutorial is a work-in-progress. We wish to improve it as soon as we can and please feel free to reach out to us if you have any questions.

{% include _toc.html %}

How many times have you analyzed a genome or metagenome and found a large proportion of the genes have no annotation?

Have you ever been interested in characterizing a bacterial genomic locus but it contains a critical gene that has no reliable homology?

Well it turns out that 40%-60% of genes predicted from microbiome have no significant homology to nucleotide databases and it's only getting worse as we sequence more and more microbiomes from around the globe! But don't worry, the	 **[AGNOSTOS-Workflow](https://github.com/functional-dark-side/agnostos-wf)** is a great way to start to address the questions above and utilize all genes, Known and Unknown, from microbiomes! ([Vanni et al., 2021](https://www.biorxiv.org/content/10.1101/2020.06.30.180448v5)) 

Recently, Vanni et al., 2021 published the preprint [Unifying the known and unknown microbial coding sequence space
](https://www.biorxiv.org/content/10.1101/2020.06.30.180448v5) showcasing the [AGNOSTOS-Workflow](https://github.com/functional-dark-side/agnostos-wf), the output of which described the unknown sequence landscape of microbiomes from the ocean and human body. Excitingly, it illuminated thousands of lineage specific genes of unknown function in the CPR superphylum Patescibacteria. To accompany the [preprint](https://www.biorxiv.org/content/10.1101/2020.06.30.180448v5), the last author [Antonio Fernandez-Guerra](http://orcid.org/0000-0002-8679-490X) wrote a blog post that goes over the origins of the [AGNOSTOS-Workflow](https://github.com/functional-dark-side/agnostos-wf) (10 years in the making), the categories of gene-clusters that result from the workflow, and the future of unknown genes in microbiomes. We highly recommend reading the [blog post](http://merenlab.org/2020/07/01/dark-side/#a-conceptual-framework-to-unify-the-known-and-the-unknown-in-microbiome-analyses) for a casual introduction into utilizing the unknown fraction genes from of microbiomes.
	
One of the goals of this [Vanni et al., 2021](https://www.biorxiv.org/content/10.1101/2020.06.30.180448v5) was to provide resources	 ([AGNOSTOS-Workflow](https://github.com/functional-dark-side/agnostos-wf) and AGNOSTOS-DB) for the science community to explore genes of unknown function in their own microbial genomic and metagenomic sequencing data. To accomplish this, we have composed the tutorial below showing how to integrate the output of the workflow from [Vanni et al., 2020](https://www.biorxiv.org/content/10.1101/2020.06.30.180448v1) into your anvi'o projects!

<div class="extra-info" markdown="1">
<span class="extra-info-header"> Here's a brief recap of the AGNOSTOS gene annotation categories discussed in Antonio's [blog post](https://merenlab.org/2020/07/01/dark-side/#a-conceptual-framework-to-unify-the-known-and-the-unknown-in-microbiome-analyses):
- **Known with Pfam annotations**: genes annotated to contain one or more Pfam entries (domain, family, repeats or motifs) but excluding the domains of unknown function (DUF)
- **Known without Pfam annotations**: which contains the genes that have a known function but lack a Pfam annotation. Here we can find intrinsically disordered proteins or small proteins among others.
- **Genomic unknown**: genes that have an unknown function (DUF are included here) and found in sequenced or draft genomes
- **Environmental unknown**: genes of unknown function not detected in sequenced or draft genomes, but only in environmental metagenomes or metagenome-assembled genomes.
</div>

The purpose of this blog post is to give an introduction into how you can utilize the unknown fraction of genes in microbiomes using the output of the [AGNOSTOS-workflow](https://github.com/functional-dark-side/agnostos-wf) in anvi'o. Throughout the tutorial we will use [the Infant Gut Dataset](https://merenlab.org/tutorials/infant-gut/) (IGD) as an example project from [Sharon et al., 2013](http://www.ncbi.nlm.nih.gov/pubmed/22936250). However, you should be able to follow these instructions to get the necessary information from an anvi'o project to annotate your genes with AGNOSTOS categories.

To start, please first follow the download instructions in the IGD tutorial [here](https://merenlab.org/tutorials/infant-gut/#downloading-the-pre-packaged-infant-gut-dataset) to get the {% include ARTIFACT name="contigs-db" text="contigs database" %} in your working directory. If you can list the following files in your working directory then you are good to follow the rest of the tutorial:

```bash
$ ls
AUXILIARY-DATA.db  CONTIGS.db  PROFILE.db  additional-files/
```

{:.warning}
If you want to go straight to data analysis, skip ahead to [Download the goods](http://localhost:4000/2021/05/28/agnostos/#download-the-goods). However, if you are interested in integrating AGNOSTOS annotations into your own anvi'o {% include ARTIFACT name="contigs-db" text="contigs database" %} then keep reading.


## Running AGNOSTOS with data from anvi'o

{:.notice}
This section is only necessary if you want to run the [AGNOSTOS-workflow](https://github.com/functional-dark-side/agnostos-wf) on your own data. If your purpose is to simply see how AGNOSTOS output can be integrated into an anvi'o project, you can skip this step and use our pre-computed output files generated by the AGNOSTOS-workflow here: [Download the goods section](http://localhost:4000/2021/05/28/agnostos/#download-the-goods). 

### Extracting data from anvi'o
	
The purpose of this section is to demonstrate how to extract the necessary data files from an anvi'o project to run the [AGNOSTOS-workflow](https://github.com/functional-dark-side/agnostos-wf). The first step is to export the gene sequences from the {% include ARTIFACT name="contigs-db" text="contigs database" %} using the anvi'o program {% include PROGRAM name="anvi-get-sequences-for-gene-calls" %}:

```bash
anvi-get-sequences-for-gene-calls -c CONTIGS.db \
                                  --get-aa-sequences \
                                  -o infant_gut_genes.fasta
```

Next, we need the gene completion information i.e. does a gene contain both a start and stop codon? This can be found in the {% include ARTIFACT name="gene-calls-txt" %} which can be extracted from your {% include ARTIFACT name="contigs-db" text="contigs database" %}. To do this, we can use the program {% include PROGRAM name="anvi-export-gene-calls" %}:

```bash
anvi-export-gene-calls -c CONTIGS.db \
                       --gene-caller prodigal \
                       -o infant_gut_gene_calls.tsv
```

And that's it!

At this point we have all the necessary files to run the [AGNOSTOS-workflow](https://github.com/functional-dark-side/agnostos-wf), namely the gene amino acid sequences (`infant_gut_genes.fasta`) and the gene completion information (`infant_gut_gene_calls.tsv`).

### Running AGNOSTOS

The purpose of this section is to demonstrate different ways of utilizing AGNOSTOS depending on the computational resources you have available and the kind of information you want to learn about your unknown sequences. The easiest way to take advantage of AGNOSTOS is to run a profile search of the AGNOSTOS-DB clusters against predicted ORFs from your sequencing data. This will quickly annotate your sequences and tell you about landscape of your unknown sequencing space. Additionally, you can leverage the metadata associated with each AGNOSTOS cluster annotation including it's lineage specificity, niche breadth, etc.

However, running the [AGNOSTOS-workflow](https://github.com/functional-dark-side/agnostos-wf) integrates your sequences into the AGNOSTOS-DB clusters or potential adds new clusters to the DB if your genomes and metagenome have novel homologous sequences (go read about that [here](https://merenlab.org/2020/07/01/dark-side/#why-gene-clusters)). The concept of ORF integration is key here because we are not passing annotations to the IGD ORFs through alignment based approaches (e.g. BLAST, hmmsearch), but are attempting to insert the IGD ORFs into the extant AGNOSTOS clusters that already have metadata associated with them (read more about it [here](http://merenlab.org/2020/07/01/dark-side/#why-gene-clusters)). Integration allows for the entire dataset of ORFs to be used rather than alignment based methods where one filter for best hits. Additionally, when IGD ORFs do no find a home in an AGNOSTOS cluster, the worflow attempts to form new clusters! 

The [AGNOSTOS-workflow](https://github.com/functional-dark-side/agnostos-wf) is a complex workflow that relies on many external dependencies. Furthermore, to achieve high levels of sensitivity, some of their steps are computationally expensive for large datasets. At the moment, The [AGNOSTOS-workflow](https://github.com/functional-dark-side/agnostos-wf), although it is fully functional on an HPC, should be considered as a proof of concept of what can be done. We are working to make it more accessible. This tutorial is the starting point for our plans to integrate an optimized version of AGNOSTOS into anvi'o.

Regardless of which way you decided to use AGNOSTOS we will be able to visualize the data in the next section.

#### Easy mode - quickly annotated sequences with AGNOSTOS categories

Here we will quickly annotate the IGD dataset with AGNOSTOS categories download the AGNOSTOS-DB profiles and cluster categories

First, download the AGNOSTOS-DB profiles and cluster categories here:

```bash
wget https://ndownloader.figshare.com/files/23066963 -O IGD_agnostos/clu_hmm_db

wget https://ndownloader.figshare.com/files/23067140 -O IGD_agnostos/cluster_ids_categ.tsv
```

Next, run the profile search.

{:.warning}
Please conda install [mmseqs2](https://github.com/soedinglab/MMseqs2#installation) to run this profile search

```bash
Profile_search/profile_search.sh --query infant_gut_genes.fasta \
			         --clu_hmm clu_hmm_db \
			         --clu_cat cluster_ids_categ.tsv \
			         --threads 10
```

#### Hard mode - integrate sequences into AGNOSTOS-DB


{:.notice}
AGNOSTOS-workflow is designed to run on an HPC environment not a standard laptop computer. Please your local bioinformatician wizard for installation

Installation instructions for the [AGNOSTOS-workflow](https://github.com/functional-dark-side/agnostos-wf) can be found [here](https://github.com/functional-dark-side/agnostos-wf/blob/master/AGNOSTOS_usage.md). Once you've completed the installation on your HPC come back here and continue the tutorial. 

Run [AGNOSTOS-workflow](https://github.com/functional-dark-side/agnostos-wf) on the IGD dataset:

```
# FIXME
a one liner from the tutorial?
```

## Download the goods


This section will go over how to integrate and visualize AGNOSTOS categories in your anvi'o projects. To have some fun, [Chiara Vanni](https://orcid.org/0000-0002-1124-1147) kindly integrated all of the ORFs from the IGD metagenomes using the [AGNOSTOS-workflow](https://github.com/functional-dark-side/agnostos-wf). Additionally she integrated the ORFs from 6 *E. faecalis* so we can investigate AGNOSTOS categories in the context of a pangenome. 

Please download these files to start the analysis:
```bash
# Make a home for the AGNOSTOS data
mkdir -p IGD_agnostos

# Download AGNOSTOS integrated IGD metagenomic
wget https://ndownloader.figshare.com/files/27896625 -O IGD_agnostos/IGD_genes_summary_info_exp.tsv

# Download AGNOSTOS integrated E. faecalis genomes
wget https://ndownloader.figshare.com/files/24028730 -O IGD_agnostos/IGD_ext_genomes_summary_info_exp.tsv
```

{:.notice}
Check out a description of the AGNOSTOS data file columns in the file [here](https://github.com/functional-dark-side/agnostos-wf/blob/master/Output_README.md).

Now you are ready to import the AGNOSTOS output for your genes back into your anvi'o project.

## Import AGNOSTOS into an anvi'o

Let's explore the AGNOSTOS integrated IGD data. The first step will be to import the data as {% include ARTIFACT name="functions" text="functions" %} into the IGD {% include ARTIFACT name="contigs-db" text="contigs database" %} (check out this [post](http://merenlab.org/2016/06/18/importing-functions/) if you have more questions about anvio functions tables). This will make it easy to explore the AGNOSTOS categories in the context of assembled contigs and read recruitment results in the anvi'o interactive interface!

First, let's import that AGNOSTOS data into the IGD {% include ARTIFACT name="contigs-db" text="contigs database" %}:
```bash
anvi-import-functions -c CONTIGS.db -p agnostos -i IGD_agnostos/IGD_genes_summary_info_exp.tsv
```


Additionally, let's import the AGNOSTOS categories into the 6 *E. faecalis* {% include ARTIFACT name="contigs-db" text="contigs databases" %} which we will use in the [Pangenomics applications](http://localhost:4000/2021/05/28/agnostos/#pangenomic-applications) section.
```bash
# FIXME: not working
for GENOME in `cut -f 2 additional-files/pangenomics/external-genomes_efaecalis.txt | tail -n +2`;
do
	fname=$(basename $GENOME .db);
	anvi-import-functions -c "${external_genomes_path}"/"${fname}".db /
						  -p agnostos /
						  -i IGD_agnostos/"${fname}"_agnostos_categories.tsv;
done
```

## Metagenomics applications

Now that we have imported the AGNOSTOS categories into our {% include ARTIFACT name="contigs-db" text="contigs database" %}, let's check out the distribution of categories in the IGD metagenome dataset:

```bash
$ tail -n+2 IGD_agnostos/IGD_genes_summary_info_exp.tsv | cut -f 7 | sort | uniq -c

2851 DISC
  75 EU
3669 GU
36284 K
1586 KWP
   9 NA
9730 SING
```


Now we can leverage these categories to explore Unknowns in metagenomic-assembled genomes (MAGs)! 

First, let's import some bins Meren made:
```
anvi-import-collection additional-files/collections/merens.txt \
                       --bins-info additional-files/collections/merens-info.txt \
                       -p PROFILE.db \
                       -c CONTIGS.db \
                       -C default
```

{:.notice}
In this analysis we are not using Singleton genes however the AGNOTOS-workflow does classify these genes. FIXME

Now because we already imported the AGNOSTOS output into anvi'o, we can immediately visualize the results like this:
```
anvi-interactive -p PROFILE.db -c CONTIGS.db -F Agnostos
```
[![agnostos_binning](/images/agnostos_binning_1.png)](){:.center-img .width-100}

[![agnostos_binning](/images/agnostos_legend.jpeg)](images/agnostos_legend.jpeg){:.center-img .width-2}






## Pangenomic applications

{:.warning}
We are working on this section! Once we are done, this section will include instructions to add proportion of AGNOSTOS categories for each gene cluster in a given anvi'o pangenome as a layer:

[![agnostos_pan](/images/agnostos_pan.png)](images/agnostos_pan.png){:.center-img .width-100}


# Conclusion

The AGNOSTOS workflow from [Vanni et al., 2020](https://www.biorxiv.org/content/10.1101/2020.06.30.180448v1) provides a path forward to utilzing all genes from microbiomes and integerates seamlessly into your standard metagenomic or pangenomic workflow as seen above using anvio. We hope this brief introduction will catalyze your future microbiomes analyses!

Please refer to our recent preprint [Light into the darkness: Unifying the known and unknown coding sequence space in microbiome analyses](https://www.biorxiv.org/content/10.1101/2020.06.30.180448v1) and [blog post](http://merenlab.org/2020/07/01/dark-side/#a-conceptual-framework-to-unify-the-known-and-the-unknown-in-microbiome-analyses) by [Antonio](http://orcid.org/0000-0002-8679-490X) for background and technical details on how to illuminate the genes of unknown function.

"To help you stop having to sweep the unknown under the carpet, we are simply removing the carpet." - Antonio


