---
layout: post
title: "An interdisciplinary Wolbachia story through connections across the Atlantic: Metagenomics, long-read sequencing, and other things"
excerpt: "After 24 months of work, our study describing the discovery of a plasmid for Wolbachia appeared in Nature Communications. I thought that this would be a good moment to take a break and think through everything that led to this proud moment and learn from it."
modified: 2019-03-05
tags: []
authors: [julie]
categories: [miscellaneous]
comments: true
image:
    feature: http://merenlab.org/images/miscellaneous/2019-03-05-wolbachia-plasmid/mbl.jpg
---

{% capture images %}{{site.url}}/images/miscellaneous/2019-03-05-wolbachia-plasmid{% endcapture %}

{:.notice}
This blog post simultaneously appeared on this page as well as [on this one](https://naturemicrobiologycommunity.nature.com/users/208603-julie-reveillaud/posts/44531-an-interdisciplinary-wolbachia-story-through-connections-across-the-atlantic-metagenomics-long-read-sequencing-and-other-things).

<em>Wolbachia</em> is a bacterium that lives inside the cells of many different insects worldwide, including mosquito. As the survival of <em>Wolbachia</em> entirely depends on their presence in a host system, they developed many strategies to spread themselves throughout the insect populations. It uses parthenogenesis, male killing, feminizing, and cytoplasmic incompatibility to make sure its population size increases. Because of this capacity to manipulate the host reproduction, and to interfere with pathogen transmission, it is one of the most promising solutions for vector control. According to the World Health Organization (WHO), mosquitoes, which are vectors of deadly pathogens such as Zika, Dengue, Chikungunya, Plasmodium, are the greatest threat to public health. The burden of these diseases has led many scientists to study the mosquito host, its immune system, but also its associated microbiota and notably <em>Wolbachia</em>. The endosymbiont is used in vast transfection programs like <a href="https://www.worldmosquitoprogram.org/">The World Mosquito Program</a> and has now become a charismatic microorganism for global health worldwide and science education through <a href="https://news.vanderbilt.edu/2019/01/22/biologist-duo-brings-wolbachia-curriculum-to-students-citizen-scientists-around-globe/">Discover the Microbes Within! The <em>Wolbachia</em> Project</a>. After the introduction of <em>Wolbachia</em> in the species <em>Aedes aegypti</em>, the program relies on the release of a limited number of mosquitoes with the newly introduced <em>Wolbachia</em> and the breeding of these with wild populations. As a result, more mosquitoes are being infected with <em>Wolbachia</em> and the transmission of pathogens viruses to local mosquito populations diminishes. Yet, the mechanisms of the interactions, at the molecular level, between the mosquito, pathogens, and <em>Wolbachia</em> are poorly known. 

Thanks to the advent of high-throughput sequencing methods, we are now capable of looking at the products of million years of evolution with new dissecting tools. It is like if we had a brand-new pair of glasses and some highly sophisticated tweezers in our hands. A gift from the past generations of scientists who made these technological developments possible. Through shotgun metagenomics (that is, the sequencing of all DNA present in an environment), assembly, binning, and genome reconstruction, we theoretically can approach the uncultivable fraction of microorganisms. We decipher the genomic potential of these different forms of Life and their intimate connection within their natural environment. Theoretically...

The more complex the microbial community, the harder it gets. And the complexity linked to endosymbionts, which live within the cells of their animal host, is an order of magnitude higher. Attempts to cultivate <em>Wolbachia</em> in laboratory settings have not been successful and getting enough pure DNA of the bacterium is a struggle. The mosquito genome is approximately 1 Giga base pair while the bacterium is around 1 Mega base pair. Host contaminations are a true bottleneck. Most metagenomic <em>Wolbachia</em> studies are done on pools of mosquitoes. Or better said, most metagenomic <em>Wolbachia</em> smartly focus on pools of embryos, which show the highest ratio of bacteria to host. But how could we tackle <em>Wolbachia</em> in its real environment. That is, a single <em>Wolbachia</em> population within the tiny reproductive organs of a tiny single mosquito. The amount of DNA extracted from the mosquito’s ovaries was less than 1 nanogram per microliter and the idea of focusing on individual mosquitoes sounded rather crazy&hellip;until I told <a href="http://merenlab.org/">Meren</a>, the last author of this study, about this new research and idea. 

Meren and I met in Woods Hole, Massachusetts, in 2011 when we were both doing our post-doctoral training at the Bay Paul Center of the <a href="https://www.mbl.edu/">Marine Biological Laboratory</a>. Meren was a computational scientist and I was coming from a molecular evolution background. In the MBL’s supportive environment, we were a small group of young and more established scientists working together on free-living and host-associated microbial ecology projects. The small community of scientists working at the MBL at that time constantly fostered spontaneous and exciting science conversations often supported by beer or coffee. These conversations sometimes turned into experiments, other times into thousands of lines of codes that gave birth to tools such <a href="http://merenlab.org/software/oligotyping/">oligotyping</a> and <a href="http://merenlab.org/software/anvio/">anvi’o</a><span class="MsoHyperlink">. While being a part of that community </span>was a unique experience, I had to start over in France to implement a new research program at <a href="http://www.inra.fr/index.php/en">INRA</a>.

[![]({{images}}/mbl.jpg)]({{images}}/mbl.jpg){:.center-img .width-90}

When one accepts a new position and starts everything from scratch, a strong network means so much. Invaluable connections between scientists give the space we need to think freely, and to dare. Despite all the sophisticated tools and technologies, one scientist alone sifting through questions about complex biological systems can quickly feel like facing Everest. The support and trust, experience and contagious excitement among few colleagues are essential to climb that mountain. Fortunately, a part of the MBL crowd kept coming together and reviving those productive MBL days annually in France, thanks to <a href="http://pagesperso.univ-brest.fr/~maignien/people.html">Loïs Maignien</a> (then a post-doc at the MBL and now a professor at the University of Brest), who has been organizing the <a href="https://twitter.com/hashtag/EBAME?src=hash">EBAME</a> workshop each year since 2014. Through EBAME we kept on staying informed about projects of each other’s in a learning environment, and even worked together, which led to our <em>Wolbachia</em> story. 

[![]({{images}}/ebame.jpg)]({{images}}/ebame.jpg){:.center-img .width-90}

Meren was convinced these pooling approaches were actually missing naturally occurring genomic diversity among <em>Wolbachia</em> in individual mosquitoes. We decided to test whether we could reconstruct <em>Wolbachia</em> genomes directly from single ovary metagenomes, and study them at the individual level through comparative genomics. The first step was to overcome technical limitations on the wet-lab side. <a href="https://www6.montpellier.inra.fr/cbgp_eng/Staff/Permanent-staff/Jousselin">Emmanuelle Jousselin</a>, who works on aphid symbiosis at INRA in Montpellier, introduced me to <a href="http://jacob.cea.fr/drf/ifrancoisjacob/Pages/Departements/Genoscope/Les-laboratoires/Laboratoire-de-sequencage.aspx?Type=Chapitre&numero=2">Corinne Cruaud</a> from the Genoscope in Evry. Corinne had recently tried a new sequencing kit that might allow constructing metagenomic libraries from very low biomass samples. She was willing to give it a try and I did my math: with the help of the C2B bioinformatic colleagues in Montpellier, we estimated that this strategy could give us 3 million reads from each <em>Wolbachia</em> population. 

So off we went; we sampled mosquitoes together with Albane Marie, Gregory Lambert and Jean-Baptiste Panchau from <a href="http://www.eid-med.org/">EID</a>. <a href="http://publications.cirad.fr/auteur.php?mat=13254">Ignace Rakotoarivony</a>, a gifted entomologist from my lab, did the microdissection and separated the ovaries from the mosquito’s body. It seemed so easy as I watched him elegantly separating almost microscopic ovaries from already very small mosquitoes. We sequenced it all, and then it was the time to deal with complex metagenomes.

[![]({{images}}/ignace.jpg)]({{images}}/ignace.jpg){:.center-img .width-50}

We assembled, re-assembled, co-assembled, single-assembled our metagenomes, until we trusted that we knew how to get most out of this particular system through genome-resolved strategies. Every attempt gave us something better and taught us more about the nature of these data. This experience was yet another demonstration that one has to play with his data in many ways and systematically. This may sound depressing, but it is also comforting to know that bioinformatics is never close to a press button solution. So the extra effort is worthwhile.

To speed up the computational work, I went to visit Meren and his group at the University of Chicago with a travel grant from the <a href="http://www.mem.inra.fr/en">INRA Metaprogram MEM</a>. Sitting next to <a href="https://twitter.com/ozcanesen">Ozcan C. Esen</a> and <a href="https://twitter.com/alon_shaiber">Alon Shaiber</a>, my computational struggles seemed to disappear magically. We automatically binned and then manually refined a <em>Wolbachia</em> genome from every mosquito metagenome we had (in fact our reproducible bioinformatics workflow is <a href="http://merenlab.org/data/2019_Reveillaud_and_Bordenstein_et_al_Wolbachia">here</a>). But even after we had enough confidence in our genome bins, some things did not add up: some parts of the <em>Wolbachia</em> genomes we recovered from different mosquitoes were consistently 5 times more abundant than the rest of the <em>Wolbachia</em> chromosome with no homology to the reference bacterial genome, its phages, nor to the mosquito host. We had come to the limits of our expertise.

[![]({{images}}/merenlab.png)]({{images}}/merenlab.png){:.center-img .width-50}

We reached out to <a href="https://twitter.com/srbordenstein">Sarah Bordenstein</a> and <a href="https://twitter.com/Symbionticism">Seth Bordenstein</a> at Vanderbilt University, two established names in the <em>Wolbachia</em> world, who also had MBL in their history. Throughout multiple Skype meetings and e-mail threads we dissected these curious contigs one by one. Until Sarah one day suggested that we might have stepped on a really fascinating discovery: a <em>Wolbachia</em> plasmid. This was surprising as plasmids in <em>Wolbachia</em> have been searched for, as these mini chromosomes would represent promising genome editing tools. Even though plasmids were found in close relatives of <em>Wolbachia</em>, such as the Rickettsia, no one had been able to identify an extra chromosomal­ element in <em>Wolbachia</em>. We were skeptical.

This was the beginning of a long series of lab and PCR experiments, more metagenomic investigations, back and forth emails, ultimately long-read sequencing and another cross Atlantic trip, this time to Nashville. Getting to discover hot chicken and live music makes you immediately fall for the Music City. We wrapped it all together and then with our colleagues back in Europe. <a href="http://www.isem.univ-montp2.fr/fr/personnel/equipes/evolution-vecteurs-adaptation-et-symbioses/weill-mylene.index/">Mylene Weill</a> from ISEM in Montpellier and <a href="https://scholar.google.co.uk/citations?user=_s58FI8AAAAJ&hl=en">Maria Fernández de Marco</a> from Animal and Plant Health Agency in UK provided some precious <em>Wolbachia</em>-free mosquito samples and helped collecting all the proofs needed for that plasmid discovery. We submitted our paper for review.

[![]({{images}}/bordensteins.jpeg)]({{images}}/bordensteins.jpeg){:.center-img .width-70}


During the review process, the excellent line of questioning from one of our referees made us realize that what we assumed to be a multi-copy extrachromosomal and circular element could have been a tandem repeat on the chromosome, and our PCR experiments may not have been enough to rule out that possibility. We needed a strategy that could give us single molecules without any amplification step to see if the ‘plasmid’ only occurs in single molecules without any upstream or downstream chromosomal DNA. We discovered the difficulties and power of the Nanopore strategy along our way. <a href="http://www.isem.univ-montp2.fr/fr/personnel/equipes/evolution-vecteurs-adaptation-et-symbioses/makoundou-patrick.index/">Patrick Makoundou</a> from ISEM shared some old school yet nowadays really precious protocols for extracting high molecular weight genomic DNA. We prepared samples for long read sequencing while <a href="https://www.researchgate.net/profile/Karen_Lolans">Karen Lolans</a> and <a href="https://twitter.com/c4vansite">Andrea Watson</a> from the University of Chicago, with their latest Nanopore experience, elegantly adapted these low biomass samples to the flow cell, and resulted in this figure that establishes it –it was after all an extrachromosomal and circular element:

[![]({{images}}/minion-metagenomics.png)]({{images}}/minion-metagenomics.png){:.center-img .width-90}


This journey taught me the frontiers between biology, mathematics, and computer science are blurring if not disappearing. This biological discovery was only possible through the contribution from people with really different yet complementary backgrounds. This is the most beautiful part of this job, I think. Like the system we study, it is a complex crosstalk yet one of the most exciting. We are like symbionts and their host, interacting and connecting to sustain ourselves. We were able to contribute a missing piece of the <em>Wolbachia</em> puzzle because we joined forces with those who shared the excitement. As people in Woods Hole say, ‘it takes a village’.


<a href="https://juliereve.wordpress.com/">Julie Reveillaud</a>, INRA Research Scientist, <a href="https://www.cirad.fr/en/our-research/research-units/animals-health-territories-risks-and-ecosystems">ASTRE Unit</a>, Montpellier, France

