---
layout: publications
modified: 2015-02-05
comments: false
---

<script type='text/javascript' src='https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js'></script>

<script async src="https://badge.dimensions.ai/badge.js" charset="utf-8"></script>

<div class="category-box">
<a href="#2019"><span class="category-item">2019 <small>(4)</small></span></a><a href="#2018"><span class="category-item">2018 <small>(6)</small></span></a><a href="#2017"><span class="category-item">2017 <small>(10)</small></span></a><a href="#2016"><span class="category-item">2016 <small>(5)</small></span></a><a href="#2015"><span class="category-item">2015 <small>(8)</small></span></a><a href="#2014"><span class="category-item">2014 <small>(10)</small></span></a><a href="#2013"><span class="category-item">2013 <small>(5)</small></span></a>
</div>

<a name="2019">&nbsp;</a>
<h1>2019</h1>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1128/mBio.00725-19"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1128/mBio.00725-19" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1128/mBio.00725-19" target="_new">Composite metagenome-assembled genomes reduce the quality of public genome repositories.</a></h3>
    <span class="pub-authors"><span class="pub-member-author">Shaiber, A.</span>, and <span class="pub-member-author">Eren, A. M</span>.</span>
    <div class="pub-info">
    <div class="pub-featured-image">
    <a href="../images/pubs/shaiber_eren_MAG_refinement.png"><img src="../images/pubs/shaiber_eren_MAG_refinement.png" style="max-width: 100px; max-height: 80px; width: auto; border: none; height: auto; margin: 0 auto; display: block; transform: translateY(15%);"/></a>
    </div>
    <div class="pub-highlights">
    <span style="display: inline-block; padding-bottom: 5px;">- A letter that stresses that the composite metagenome-assembled genomes influence phylogenomic, pangenomic, and ecological insights (<a href="../files/Shaiber_Eren_2019_reviewer_response.pdf" target="_blank">peer reviews and responses</a>).</span><br><span style="display: inline-block; padding-bottom: 5px;">-  <a href="http://merenlab.org/data/refining-espinoza-mags/" target="_blank">A reproducible workflow</a> to detail the steps of genome refinement and make available the refined versions of some key genomes.</span>
    </div>
    </div>
    <span class="pub-journal"><i>mBio</i>. <b>10(3) e00725-19</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1038/s41467-019-08973-w"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1038/s41467-019-08973-w" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1038/s41467-019-08973-w" target="_new">The <i>Wolbachia</i> mobilome in <i>Culex pipiens</i> includes a putative plasmid.</a></h3>
    <span class="pub-authors">Reveillaud, R., Bordenstein, S. R., Cruaud, C., <span class="pub-member-author">Shaiber, A.</span>, <span class="pub-member-author">Esen, Ö. C.</span>, O. C., Weill, M., Makoundou, P., <span class="pub-member-author">Lolans, K.</span>, <span class="pub-member-author">Watson, A. R.</span>, Rakotoarivony, I., Bordenstein, S. R., and <span class="pub-member-author">Eren, A. M</span>.</span>
    <div class="pub-info">
    <div class="pub-featured-image">
    <a href="../images/miscellaneous/2019-03-05-wolbachia-plasmid/minion-metagenomics.png"><img src="../images/miscellaneous/2019-03-05-wolbachia-plasmid/minion-metagenomics.png" style="max-width: 100px; max-height: 80px; width: auto; border: none; height: auto; margin: 0 auto; display: block; transform: translateY(15%);"/></a>
    </div>
    <div class="pub-highlights">
    <span style="display: inline-block; padding-bottom: 5px;">- The first report of a <i>Wolbachia</i> plasmid through genome-resolved metagenomics on microsurgically removed individual mosquito ovary samples (<a href="https://static-content.springer.com/esm/art%3A10.1038%2Fs41467-019-08973-w/MediaObjects/41467_2019_8973_MOESM2_ESM.pdf">peer reviews and responses</a>).</span><br><span style="display: inline-block; padding-bottom: 5px;">-  Yet another application of <a href="https://doi.org/10.7717/peerj.4320">metapangenomics</a> and an applicatoin of minION long-read sequencing on extremely low-biomass samples.</span><br><span style="display: inline-block; padding-bottom: 5px;">-  <a href="http://merenlab.org/data/wolbachia-plasmid">Reproducible bioinformatics workflow</a> with all data items, a 'behind the paper' <a href="http://merenlab.org/2019/03/05/wolbachia-plasmid/">blog post</a> by <a href="https://twitter.com/JulieReve">Julie Reveillaud</a>, and a <a href="https://www.mbl.edu/blog/new-mobile-element-found-in-mosquito-parasite-has-potential-for-disease-control/">press release</a> from the Marine Biological Laboratory.</span>
    </div>
    </div>
    <span class="pub-journal"><i>Nature Communications</i>. <b>10:1051</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1038/s41564-019-0362-4"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1038/s41564-019-0362-4" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1038/s41564-019-0362-4" target="_new">Co-occurring genomic capacity for anaerobic methane and dissimilatory sulfur metabolisms discovered in the Korarchaeota.</a></h3>
    <span class="pub-authors">McKay, L. J., Dlakić, M., Fields, M. W., <span class="pub-member-author">Delmont, T. O.</span>, <span class="pub-member-author">Eren, A. M</span>., Jay, Z. J., Klingelsmith, K. B., Rusch, D. B., and Inskeep, W. P.</span>
    <div class="pub-info">
    <div class="pub-featured-image">
    <a href="../images/pubs/mckay_and_inskeep.jpg"><img src="../images/pubs/mckay_and_inskeep.jpg" style="max-width: 100px; max-height: 80px; width: auto; border: none; height: auto; margin: 0 auto; display: block; transform: translateY(15%);"/></a>
    </div>
    <div class="pub-highlights">
    <span style="display: inline-block; padding-bottom: 5px;">- Yesllowstone National Park, archaeal evolution, genome-resolved metagenomics, phylogenomics, and pangenomics (cool stuff all over).</span><br><span style="display: inline-block; padding-bottom: 5px;">-  First description of a microbial population with both anaerobic methane and dissimilatory sulfur metabolisms.</span><br><span style="display: inline-block; padding-bottom: 5px;">-  A news article by Montana State University with the photographs of cools scientists: <a href="http://www.montana.edu/news/18471">An organism in Yellowstone hot spring potentially linked to earliest life on Earth"</a></span>
    </div>
    </div>
    <span class="pub-journal"><i>Nature Microbiology</i>. <b>2058-5276</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1038/s41564-019-0494-6"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1038/s41564-019-0494-6" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1038/s41564-019-0494-6" target="_new">Global phylogeography and ancient evolution of the widespread human gut virus crAssphage.</a></h3>
    <span class="pub-authors">Edwards, R. A., Vega, A. A., Norman, H. M., Ohaeri, M., Levi, K., Dinsdale, E. A., Cinek, O., Aziz, R. K., McNair, K., Barr, J. J., Bibby, K., Brouns, SJJ., Cazares, A., deJonge, P. A., Desnues, C., Díaz Muñoz, S. L., Fineran, P. C., Kurilshikov, A., Lavigne, R., Mazankova, K., McCarthy, D. T., Nobrega, F. L., Reyes Muñoz, A., Tapia, G., Trefault, N., Tyakht, A. V., Vinuesa, P., Wagemans, J., Zhernakova, A., Aarestrup, F. M., Ahmadov, G., Alassaf, A., Anton, J., Asangba, A., Billings, E. K., Cantu, V. A., Carlton, J. M., Cazares, D., Cho, G. S., Condeff, T., Cortés, P., Cranfield, M., Cuevas, D. A., De la Iglesia, R., Decewicz, P., Doane, M. P., Dominy, N. J., Dziewit, L., Elwasila, B. M., <span class="pub-member-author">Eren, A. M</span>., Franz, C., Fu, J., Garcia-Aljaro, C., Ghedin, E., Gulino, K. M., Haggerty, J. M., Head, S. R., Hendriksen, R. S., Hill, C., Hyöty, H., Ilina, E. N., Irwin, M. T., Jeffries, T. C., Jofre, J., Junge, R. E., Kelley, S. T., Khan Mirzaei, M., Kowalewski, M., Kumaresan, D., Leigh, S. R., Lipson, D., Lisitsyna, E. S., Llagostera, M., Maritz, J. M., Marr, L. C., McCann, A., Molshanski-Mor, S., Monteiro, S., Moreira-Grez, B., Morris, M., Mugisha, L., Muniesa, M., Neve, H., Nguyen, N. P., Nigro, O. D., Nilsson, A. S., O'Connell, T., Odeh, R., Oliver, A., Piuri, M., Prussin Ii, A. J., Qimron, U., Quan, Z. X., Rainetova, P., Ramírez-Rojas, A., Raya, R., Reasor, K., Rice, GAO., Rossi, A., Santos, R., Shimashita, J., Stachler, E. N., Stene, L. C., Strain, R., Stumpf, R., Torres, P. J., Twaddle, A., Ugochi Ibekwe, M., Villagra, N., Wandro, S., White, B., Whiteley, A., Whiteson, K. L., Wijmenga, C., Zambrano, M. M., Zschach, H., and Dutilh, B. E.</span>
    <div class="pub-info">
    <div class="pub-featured-image">
    <a href="../images/pubs/crAssphage.png"><img src="../images/pubs/crAssphage.png" style="max-width: 100px; max-height: 80px; width: auto; border: none; height: auto; margin: 0 auto; display: block; transform: translateY(15%);"/></a>
    </div>
    <div class="pub-highlights">
    <span style="display: inline-block; padding-bottom: 5px;">- A monumental effort lead by <a href="https://twitter.com/linsalrob" target="_blank">Rob Edwards</a> and <a href="https://twitter.com/BEDutilh" target="_blank">Bas Dutilh</a> to explore with the help of 114 scientists from around the world the global phylogeorgraphy and evolution of <a href="https://www.nature.com/articles/ncomms5498" target="_blank">crAssphage</a>, one of the most numerous viruses in the human gut that infect bacteria.</span><br><span style="display: inline-block; padding-bottom: 5px;">- A curated list of press coverage of this study is available <a href="https://edwards.sdsu.edu/research/press-about-global-phylogeography-of-crassphage/" target="_blank">on Rob Edwards' web site</a>.</span>
    </div>
    </div>
    <span class="pub-journal"><i>Nature Microbiology</i>. <b>ahead of print</b></span>
</div>


<a name="2018">&nbsp;</a>
<h1>2018</h1>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1038/s41467-018-07675-z"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1038/s41467-018-07675-z" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1038/s41467-018-07675-z" target="_new">Microbiome characterization by high-throughput transfer RNA sequencing and modification analysis.</a></h3>
    <span class="pub-authors">Schwartz, M. H., Wang, H., Pan, J. N., Clark. W. C., <span class="pub-member-author">Cui, S.</span>, Eckwahl, M. J., Pan, D. W., Parisien, M., Owens, S. M., Cheng, B. L., Martinez, K., Xu, J., Chang, E. B., Pan, T., and <span class="pub-member-author">Eren, A. M</span>.</span>
    <div class="pub-info">
    <div class="pub-featured-image">
    <a href="../images/pubs/schwartz_et_al_trna_seq.png"><img src="../images/pubs/schwartz_et_al_trna_seq.png" style="max-width: 100px; max-height: 80px; width: auto; border: none; height: auto; margin: 0 auto; display: block; transform: translateY(15%);"/></a>
    </div>
    <div class="pub-highlights">
    <span style="display: inline-block; padding-bottom: 5px;">- First application of tRNA sequencing to microbiomes (<a href="https://static-content.springer.com/esm/art%3A10.1038%2Fs41467-018-07675-z/MediaObjects/41467_2018_7675_MOESM2_ESM.pdf">peer reviews and responses</a>).</span><br><span style="display: inline-block; padding-bottom: 5px;">-  tRNA-seq distinguishes high-fat- and low-fat-fed mice in a comparable fashion to 16S rRNA gene amplicons, reveals taxon- and diet-dependent variations in tRNA modifications, and provides in situ insights into tRNA gene expression dynamics and post-transcriptional modifications.</span><br><span style="display: inline-block; padding-bottom: 5px;">-  "<a href="https://www.uchicagomedicine.org/forefront/microbiome-articles/2018/december/new-rna-sequencing-strategy-provides-insight-into-microbiomes">New RNA sequencing strategy provides insight into microbiomes</a>", by <b>Matt Wood</b>.</span><br><span style="display: inline-block; padding-bottom: 5px;">-  "<a href="https://www.news-medical.net/news/20181217/RNA-sequencing-offers-novel-insights-into-the-microbiome.aspx">RNA sequencing offers novel insights into the microbiome</a>", by <b>Liji Thomas</b>, MD, and <b>Kate Anderton</b>, B.Sc.</span>
    </div>
    </div>
    <span class="pub-journal"><i>Nature Communications</i>. <b>9:5353</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1038/s41564-018-0176-9"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1038/s41564-018-0176-9" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1038/s41564-018-0176-9" target="_new">Nitrogen-fixing populations of Planctomycetes and Proteobacteria are abundant in the surface ocean metagenomes.</a></h3>
    <span class="pub-authors"><span class="pub-member-author">Delmont, T. O.</span>, Quince, C., <span class="pub-member-author">Shaiber, A.</span>, <span class="pub-member-author">Esen, Ö. C.</span>, <span class="pub-member-author">Lee, S. T. M.</span>, Rappé, M. S., McLellan, S. L., Lücker, S., and <span class="pub-member-author">Eren, A. M</span>.</span>
    <div class="pub-info">
    <div class="pub-featured-image">
    <a href="http://i.imgur.com/oFRefc4.jpg"><img src="http://i.imgur.com/oFRefc4.jpg" style="max-width: 100px; max-height: 80px; width: auto; border: none; height: auto; margin: 0 auto; display: block; transform: translateY(15%);"/></a>
    </div>
    <div class="pub-highlights">
    <span style="display: inline-block; padding-bottom: 5px;">- First genomic evidence for abundant and widespread non-cyanobacterial nitrogen-fixing populations in the surface ocean.</span><br><span style="display: inline-block; padding-bottom: 5px;">-  Nearly 1,000 non-redundant, high-quality bacterial, archaeal, and eukaryotic population genomes from TARA Oceans metagenomes.</span><br><span style="display: inline-block; padding-bottom: 5px;">-  A "behind the paper" <a href="https://naturemicrobiologycommunity.nature.com/users/113363-a-murat-eren-meren/posts/34040-microbiologists-vs-shotgun-metagenomes-surface-ocean" target="_blank">blog post</a> by Meren, a <a href="http://www.mbl.edu/blog/large-scale-study-indicates-novel-and-abundant-nitrogen-fixing-microbes-in-the-surface-ocean/" target="_blank">press release</a> by the MBL, and an extensive description of <a href="http://merenlab.org/data/2017_Delmont_et_al_HBDs/" target="_blank">the bioinformatics workflow</a>.</span>
    </div>
    </div>
    <span class="pub-journal"><i>Nature Microbiology</i>. <b>3:804–813</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.7717/peerj.4320"></div>
<div class="__dimensions_badge_embed__" data-doi="10.7717/peerj.4320" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.7717/peerj.4320" target="_new">Linking pangenomes and metagenomes: the <i>Prochlorococcus</i> metapangenome.</a></h3>
    <span class="pub-authors"><span class="pub-member-author">Delmont, T. O.</span>, and <span class="pub-member-author">Eren, A. M</span>.</span>
    <div class="pub-info">
    <div class="pub-featured-image">
    <a href="https://i.imgur.com/gzDpEU5.jpg"><img src="https://i.imgur.com/gzDpEU5.jpg" style="max-width: 100px; max-height: 80px; width: auto; border: none; height: auto; margin: 0 auto; display: block; transform: translateY(15%);"/></a>
    </div>
    <div class="pub-highlights">
    <span style="display: inline-block; padding-bottom: 5px;">- A big-data study in which a pangenome of 31 <i>Prochlorococcus</i> isolates meets 31 billion Tara Oceans metagenomic sequences (<a href="https://peerj.com/articles/4320/reviews/">Peer-review history</a>).</span><br><span style="display: inline-block; padding-bottom: 5px;">-  Metapangenomes reveal to what extent genes that may be linked to the ecology and fitness of microbes are conserved within a phylogenetic clade.</span><br><span style="display: inline-block; padding-bottom: 5px;">-  <a href="http://merenlab.org/data/prochlorococcus-metapangenome" target="_blank">Reproducible bioinformatics workflow</a>.</span>
    </div>
    </div>
    <span class="pub-journal"><i>PeerJ</i>. <b>6:e4320</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1038/s41422-018-0127-2"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1038/s41422-018-0127-2" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1038/s41422-018-0127-2" target="_new">Transcriptome-wide reprogramming of N<sup>6</sup>-methyladenosine modification by the mouse microbiome.</a></h3>
    <span class="pub-authors">Wang, X., Li, Y., Chen, W., Shi, H., <span class="pub-member-author">Eren, A. M</span>., Morozov, A., He, C., Luo, G. Z., and Pan, T.</span>
    <div class="pub-info">
    <div class="pub-featured-image">
    <a href="../images/pubs/wang_et_al_m6A.png"><img src="../images/pubs/wang_et_al_m6A.png" style="max-width: 100px; max-height: 80px; width: auto; border: none; height: auto; margin: 0 auto; display: block; transform: translateY(15%);"/></a>
    </div>
    <div class="pub-highlights">
    <span style="display: inline-block; padding-bottom: 5px;">-  N<sup>6</sup>-methyladenosine (m<sup>6</sup>A) is the most abundant mesenger RNA modification in mammalian cells, occurring at ~3 modified adenosine residues per transcript.</span><br><span style="display: inline-block; padding-bottom: 5px;">-  Using liquid chromatography/mass spectrometry, this study shows differential occurrence of m<sup>6</sup>A modifications in brain, liver, and intestinal cells between germ-free and conventional mice, demonstrating that the microbiome has a strong effect on host m<sup>6</sup>A mRNA modification.</span>
    </div>
    </div>
    <span class="pub-journal"><i>Cell Research</i>. <b>0:1-4</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1186/s40168-018-0474-8"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1186/s40168-018-0474-8" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1186/s40168-018-0474-8" target="_new">Gut microbes contribute to variation in solid organ transplant outcomes in mice.</a></h3>
    <span class="pub-authors">McIntosh, C. M., Chen, L., <span class="pub-member-author">Shaiber, A.</span>, <span class="pub-member-author">Eren, A. M</span>., and Alegre, M. L.</span>
    <div class="pub-info-no-image">
    <div class="pub-highlights-no-image">
    <span style="display: inline-block; padding-bottom: 5px;">- Two groups of genetically identical mice with different microbial community structures differ in their response to skin transplants from genetically identical sources: one group rejects the skin faster than the other.</span><br><span style="display: inline-block; padding-bottom: 5px;">-  Transferring poop from slow-rejecting mice into fast-rejecting mice turns fast-rejecting mice into slow-rejecting mice.</span><br><span style="display: inline-block; padding-bottom: 5px;">-  These suggest that differences in 'resident microbiome' in healthy individuals <i>may</i> contribute to interpersonal variability in graft outcomes, and fecal microbiota transplantation <i>may</i> play a therapeutic role to reverse that phenotype.</span><br><span style="display: inline-block; padding-bottom: 5px;">-  Basically, "we do not have a mechanistic understanding, but this is definitely very interesting".</span>
    </div>
    </div>
    <span class="pub-journal"><i>Microbiome</i>. <b>6:96</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1038/s41586-018-0125-z"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1038/s41586-018-0125-z" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1038/s41586-018-0125-z" target="_new">Microbial signals drive pre-leukaemic myeloproliferation in a Tet2-deficient host.</a></h3>
    <span class="pub-authors">Meisel, M., Hinterleitner, R., Pacis, A., Chen, L., Earley, Z. M., Mayassi, T., Pierre, J. F., Ernest, J. D., Galipeau, H. J., Thuille, N., Bouziat, R., Buscarlet, M., Ringus, D. L., Wang, Y., Li, Y., Dinh, V., Kim, S. M., McDonald, B. D., Zurenski, M. A., Musch, M. W., Furtado, G. C., Lira, S. A., Baier, G., Chang, E. B., <span class="pub-member-author">Eren, A. M</span>., Weber, C. R., Busque, L., Godley, L. A., Verdú, E. F., Barreiro, L. B., and Jabri, B.</span>
    <div class="pub-info-no-image">
    <div class="pub-highlights-no-image">
    <span style="display: inline-block; padding-bottom: 5px;">- TET2 deficiency ➡️  deteriorating small intestinal barrier ➡️  bacterial translocation ➡️  incrased IL-6 signalling ➡️  pre-leukemic myeloproliferation (a leukemia precursor).</span><br><span style="display: inline-block; padding-bottom: 5px;">-  ScienceDaily: <a href="https://www.sciencedaily.com/releases/2018/05/180516131226.htm" target="_blank">Under certain conditions, bacterial signals set the stage for leukemia</a>.</span>
    </div>
    </div>
    <span class="pub-journal"><i>Nature</i>. <b>557:580–584</b></span>
</div>


<a name="2017">&nbsp;</a>
<h1>2017</h1>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1186/s40168-017-0270-x"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1186/s40168-017-0270-x" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1186/s40168-017-0270-x" target="_new">Tracking microbial colonization in fecal microbiota transplantation experiments via genome-resolved metagenomics.</a></h3>
    <span class="pub-authors"><span class="pub-member-author">Lee, S. T. M.</span>, Kahn, S. A., <span class="pub-member-author">Delmont, T. O.</span>, <span class="pub-member-author">Shaiber, A.</span>, <span class="pub-member-author">Esen, Ö. C.</span>, Hubert, N. A., Morrison, H. G., Antonopoulos, D. A., Rubin, D. T., and <span class="pub-member-author">Eren, A. M</span>.</span>
    <div class="pub-info">
    <div class="pub-featured-image">
    <a href="http://i.imgur.com/4e4Ic4u.jpg"><img src="http://i.imgur.com/4e4Ic4u.jpg" style="max-width: 100px; max-height: 80px; width: auto; border: none; height: auto; margin: 0 auto; display: block; transform: translateY(15%);"/></a>
    </div>
    <div class="pub-highlights">
    <span style="display: inline-block; padding-bottom: 5px;">- An FMT study with metagenome-assembled genomes (see <a href="http://merenlab.org/data/#lee-et-al-2017--fecal-transplants" target="_blank">public data</a>).</span><br><span style="display: inline-block; padding-bottom: 5px;">- Bacteroidales: high-colonization rate. Clostridiales: low colonization rate. Colonization success is negatively correlated with the number of genes related to sporulation.</span><br><span style="display: inline-block; padding-bottom: 5px;">-  MAGs with the same taxonomy showed different colonization properties, highlighting the importance of high-resolution analyses.</span><br><span style="display: inline-block; padding-bottom: 5px;">-  Populations colonized both recipients were also prevalent in the HMP cohort (and the ones that did not, distribute sporadically across the HMP cohort).</span>
    </div>
    </div>
    <span class="pub-journal"><i>Microbiome</i>. <b>5:50</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1073/pnas.1712186114"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1073/pnas.1712186114" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1073/pnas.1712186114" target="_new">Simulations predict microbial responses in the environment? This environment disagrees retrospectively.</a></h3>
    <span class="pub-authors"><span class="pub-member-author">Delmont, T. O.</span>, and <span class="pub-member-author">Eren, A. M</span>.</span>
    <div class="pub-info">
    <div class="pub-featured-image">
    <a href="http://merenlab.org/images/miscellaneous/2017-10-16-reply-to-probst-et-al/Figure_01.png"><img src="http://merenlab.org/images/miscellaneous/2017-10-16-reply-to-probst-et-al/Figure_01.png" style="max-width: 100px; max-height: 80px; width: auto; border: none; height: auto; margin: 0 auto; display: block; transform: translateY(15%);"/></a>
    </div>
    <div class="pub-highlights">
    <span style="display: inline-block; padding-bottom: 5px;">- A letter that re-analyzes some of the findings published in <a href="http://www.pnas.org/content/114/28/7432.abstract" target="_blank">Hu et al.</a></span><br><span style="display: inline-block; padding-bottom: 5px;">-  Here is a <a href="http://www.pnas.org/content/early/2017/10/09/1712466114.extract" target="_blank">rebuttal from Probst et al.</a> challenging our findings in this letter.</span><br><span style="display: inline-block; padding-bottom: 5px;">-  Here is <a href="http://merenlab.org/2017/10/16/reply-to-probst-et-al/" target="_blank">our response to Probst et al.</a>, and <a href="http://merenlab.org/2017/11/25/DWH-O-desum-v2/" target="_blank">the recovery of DWH O. Desum v2</a>.</span>
    </div>
    </div>
    <span class="pub-journal"><i>Proc Natl Acad Sci U S A</i>. <b>114(43):E8947-E8949</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1038/s41467-017-01228-6"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1038/s41467-017-01228-6" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1038/s41467-017-01228-6" target="_new">Genomic variation in microbial populations inhabiting the marine subseafloor at deep-sea hydrothermal vents.</a></h3>
    <span class="pub-authors">Anderson, R. E., Reveillaud, J., Reddington, E., <span class="pub-member-author">Delmont, T. O.</span>, <span class="pub-member-author">Eren, A. M</span>., McDermott, J. M., Seewald, J. S., and Huber, J. A.</span>
    <span class="pub-journal"><i>Nature Communications</i>. <b>8:1114</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1101/170639"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1101/170639" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1101/170639" target="_new">The global biogeography of amino acid variants within a single SAR11 population is governed by natural selection.</a></h3>
    <span class="pub-authors"><span class="pub-member-author">Delmont, T. O.</span>, <span class="pub-member-author">Kiefl, E.</span>, Kilinc, O., <span class="pub-member-author">Esen, Ö. C.</span>, Uysal, I., Rappé, M. S., Giovannoni, S., and <span class="pub-member-author">Eren, A. M</span>.</span>
    <div class="pub-info">
    <div class="pub-featured-image">
    <a href="http://i.imgur.com/HrlXPOF.jpg"><img src="http://i.imgur.com/HrlXPOF.jpg" style="max-width: 100px; max-height: 80px; width: auto; border: none; height: auto; margin: 0 auto; display: block; transform: translateY(15%);"/></a>
    </div>
    <div class="pub-highlights">
    <span style="display: inline-block; padding-bottom: 5px;">- Using single-amino acid variants to investigate genome variation within a single SAR11 population across large geographies.</span><br><span style="display: inline-block; padding-bottom: 5px;">-  Application of a novel Deep Learning algorithm to resolve relationships between metagenomes through complex SAAV data.</span><br><span style="display: inline-block; padding-bottom: 5px;">-  Partitioning of SAAVs suggests natural selection, rather than neutral evolution, is the main driver of the evolution of SAR11.</span>
    </div>
    </div>
    <span class="pub-journal"><i>BioRxiv</i>. <b>pre-print</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1038/ismej.2017.165"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1038/ismej.2017.165" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1038/ismej.2017.165" target="_new">Recurrent patterns of microdiversity in a temperate coastal marine environment.</a></h3>
    <span class="pub-authors">Chafee, M., Fernàndez-Guerra, A., Buttigieg, P. L., Gerdts, G., <span class="pub-member-author">Eren, A. M</span>., Teeling, H., and Amann, R. I.</span>
    <span class="pub-journal"><i>ISME J</i>. <b>12, 237–252</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1186/s13059-017-1309-9"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1186/s13059-017-1309-9" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1186/s13059-017-1309-9" target="_new">DESMAN: a new tool for de novo extraction of strains from metagenomes.</a></h3>
    <span class="pub-authors">Quince, C., <span class="pub-member-author">Delmont, T. O.</span>, Raguideau, S., Alneberg, J., Darling, A. E., Collins, G., and <span class="pub-member-author">Eren, A. M</span>.</span>
    <span class="pub-journal"><i>Genome Biology</i>. <b>18:181</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1038/nbt.3893"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1038/nbt.3893" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1038/nbt.3893" target="_new">Minimum information about a single amplified genome (MISAG) and a metagenome-assembled genome (MIMAG) of bacteria and archaea.</a></h3>
    <span class="pub-authors">Bowers, R. M., Kyrpides, N. C., Stepanauskas, R., Harmon-Smith, M., Doud, D., Reddy, TBK., Schulz, F., Jarett, J., Rivers, A. R., Eloe-Fadrosh, E. A., Tringe, S. G., Ivanova, N. N., Copeland, A., Clum, A., Becraft, E. D., Malmstrom, R. R., Birren, B., Podar, M., Bork, P., Weinstock, G. M., Garrity, G. M., Dodsworth, J. A., Yooseph, S., Sutton, G., Glöckner, F. O., Gilbert, J. A., Nelson, W. C., Hallam, S. J., Jungbluth, S. P., Ettema, G., Tighe, S., Konstantinidis, K. T., Liu, W., Baker, B. J., Rattei, T., Eisen, J. A., Hedlund, B., McMahon, K. D., Fierer, N., Knight, R., Finn, R., Cochrane, G., Karsch-Mizrachi, I., Tyson, G. W., Rinke, C., Lapidus, A., Meyer, F., Yilmaz, P., Parks, D. H., <span class="pub-member-author">Eren, A. M</span>., Schriml, L., Banfield, J. F., Hugenholtz, P., Woyke, T.</span>
    <span class="pub-journal"><i>Nature Biotechnology</i>. <b>35:725–731</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1016/j.celrep.2017.06.060"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1016/j.celrep.2017.06.060" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1016/j.celrep.2017.06.060" target="_new">Peripartum Antibiotics Promote Gut Dysbiosis, Loss of Immune Tolerance, and Inflammatory Bowel Disease in Genetically Prone Offspring.</a></h3>
    <span class="pub-authors">Miyoshi, J., Bobe, A. M., Miyoshi, S., Huang, Y., Hubert, N., <span class="pub-member-author">Delmont, T. O.</span>, <span class="pub-member-author">Eren, A. M</span>., Leone, V., and Chang, E. B.</span>
    <div class="pub-info">
    <div class="pub-featured-image">
    <a href="http://i.imgur.com/jvmaUfV.png"><img src="http://i.imgur.com/jvmaUfV.png" style="max-width: 100px; max-height: 80px; width: auto; border: none; height: auto; margin: 0 auto; display: block; transform: translateY(15%);"/></a>
    </div>
    <div class="pub-highlights">
    <span style="display: inline-block; padding-bottom: 5px;">- Antibiotics during pregnancy promote offspring gut dysbiosis, immune dysfunction, and IBD.</span><br><span style="display: inline-block; padding-bottom: 5px;">-  Antibiotics given after the developmental period do not increase IBD.</span><br><span style="display: inline-block; padding-bottom: 5px;">-  Antibiotic-perturbed maternal microbiota likely contribute to neonatal gut dysbiosis.</span><br><span style="display: inline-block; padding-bottom: 5px;">-  <a href="https://sciencelife.uchospitals.edu/2017/07/11/antibiotics-taken-late-in-pregnancy-can-increase-risk-for-inflammatory-bowel-diseases-in-offspring/" target="_blank">Press release</a>.</span>
    </div>
    </div>
    <span class="pub-journal"><i>Cell Reports</i>. <b>20:2</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1097/MIB.0000000000001027"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1097/MIB.0000000000001027" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1097/MIB.0000000000001027" target="_new">Early Transcriptomic Changes in the Ileal Pouch Provide Insight into the Molecular Pathogenesis of Pouchitis and Ulcerative Colitis.</a></h3>
    <span class="pub-authors">Huang, Y., Dalal, S., Antonopoulos, D., Hubert, N., Raffals, L. H., Dolan, K., Weber, C., Messer, J. S., Jabri, B., Bendelac, A., <span class="pub-member-author">Eren, A. M</span>., Rubin, D. T., Sogin, M. L., and Chang, E. B.</span>
    <span class="pub-journal"><i>Inflammatory Bowel Disiseases</i>. <b>23(3):366-378</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.3389/fmicb.2017.00264"></div>
<div class="__dimensions_badge_embed__" data-doi="10.3389/fmicb.2017.00264" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.3389/fmicb.2017.00264" target="_new">A Few Pseudomonas Oligotypes Dominate in the Meat and Dairy Processing Environment.</a></h3>
    <span class="pub-authors">Stellato, G., Utter, D. R., Voorhis, A., De Angelis, M., <span class="pub-member-author">Eren, A. M</span>., and Ercolini, D.</span>
    <span class="pub-journal"><i>Front Microbiol</i>. <b>8:264</b></span>
</div>


<a name="2016">&nbsp;</a>
<h1>2016</h1>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.7717/peerj.1839"></div>
<div class="__dimensions_badge_embed__" data-doi="10.7717/peerj.1839" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.7717/peerj.1839" target="_new">Identifying contamination with advanced visualization and analysis practices: metagenomic approaches for eukaryotic genome assemblies.</a></h3>
    <span class="pub-authors"><span class="pub-member-author">Delmont, T. O.</span>, and <span class="pub-member-author">Eren, A. M</span>.</span>
    <div class="pub-info">
    <div class="pub-featured-image">
    <a href="../images/pubs/delmont_tardigrade.png"><img src="../images/pubs/delmont_tardigrade.png" style="max-width: 100px; max-height: 80px; width: auto; border: none; height: auto; margin: 0 auto; display: block; transform: translateY(15%);"/></a>
    </div>
    <div class="pub-highlights">
    <span style="display: inline-block; padding-bottom: 5px;">- A holistic approach to visualize and curate genomic and metagenomic assemblies.</span><br><span style="display: inline-block; padding-bottom: 5px;">- A re-analysis of the first released Tardigrade genome reveals a likely symbiont among other contaminants.</span><br><span style="display: inline-block; padding-bottom: 5px;">-  A practical approach to estimate the number bacterial genomes in an assembly.</span>
    </div>
    </div>
    <span class="pub-journal"><i>PeerJ</i>. <b>4:e1839</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1128/mBio.01713-16"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1128/mBio.01713-16" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1128/mBio.01713-16" target="_new">Patient-specific <i>Bacteroides</i> genome variants in pouchitis.</a></h3>
    <span class="pub-authors">Vineis, J. H., Ringus, D. L., Morrison, H. G., <span class="pub-member-author">Delmont, T. O.</span>, Dalal, S., Raffals, L. H., Antonopoulos, D. A., Rubin, D. T., <span class="pub-member-author">Eren, A. M</span>., Chang, E. B., and Sogin, M. L</span>
    <span class="pub-journal"><i>mBio</i>. <b>7(6):e01713-16</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.3389/fmicb.2016.01318"></div>
<div class="__dimensions_badge_embed__" data-doi="10.3389/fmicb.2016.01318" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.3389/fmicb.2016.01318" target="_new">New insights into microbial ecology through subtle nucleotide variation.</a></h3>
    <span class="pub-authors"><span class="pub-member-author">Eren, A. M</span>., Sogin, M. L., and Maignien, L.</span>
    <span class="pub-journal"><i>Front Microbiol</i>. <b>7:1318</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1093/femsre/fuw017"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1093/femsre/fuw017" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1093/femsre/fuw017" target="_new">Millions of reads, thousands of taxa: microbial community structure and associations analyzed via marker genes.</a></h3>
    <span class="pub-authors">Bálint, M., Bahram, M., <span class="pub-member-author">Eren, A. M</span>., Faust, K., Fuhrman, J. A., Lindahl, B., O'Hara, R. B., Öpik, M., Sogin, M. L., Unterseher, M., and Tedersoo, L.</span>
    <span class="pub-journal"><i>FEMS Microbiology Reviews</i>. <b>na:na</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1186/s40168-016-0161-6"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1186/s40168-016-0161-6" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1186/s40168-016-0161-6" target="_new">Bacterial communities in penile skin, male urethra, and vaginas of heterosexual couples with and without bacterial vaginosis.</a></h3>
    <span class="pub-authors">Zozaya, M., Ferris, M. J., Siren, J. D., Lillis, R., Myers, L., Nsuami, M. J., <span class="pub-member-author">Eren, A. M</span>., Brown, J., Taylor, C. M., and Martin, D. H.</span>
    <span class="pub-journal"><i>Microbiome</i>. <b>4:16</b></span>
</div>


<a name="2015">&nbsp;</a>
<h1>2015</h1>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.7717/peerj.1319"></div>
<div class="__dimensions_badge_embed__" data-doi="10.7717/peerj.1319" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.7717/peerj.1319" target="_new">Anvi’o: An advanced analysis and visualization platform for ‘omics data.</a></h3>
    <span class="pub-authors"><span class="pub-member-author">Eren, A. M</span>., <span class="pub-member-author">Esen, Ö. C.</span>, Quince, C., Vineis, J. H., Morrison, H. G., Sogin, M. L., and <span class="pub-member-author">Delmont, T. O.</span></span>
    <div class="pub-info">
    <div class="pub-featured-image">
    <a href="http://i.imgur.com/d1c7bUY.png"><img src="http://i.imgur.com/d1c7bUY.png" style="max-width: 100px; max-height: 80px; width: auto; border: none; height: auto; margin: 0 auto; display: block; transform: translateY(15%);"/></a>
    </div>
    <div class="pub-highlights">
    <span style="display: inline-block; padding-bottom: 5px;">- The methods paper for <a href="http://merenlab.org/software/anvio/" target="_blank">anvi'o</a>.</span><br><span style="display: inline-block; padding-bottom: 5px;">-  Binning, and single-nucleotide variant analysis of a human gut time series metagenome.</span><br><span style="display: inline-block; padding-bottom: 5px;">-  Re-analysis of cultivar genomes, metagenomes, and metatranscriptomes associated with the Deepwater Horizon oil spill.</span>
    </div>
    </div>
    <span class="pub-journal"><i>PeerJ</i>. <b>6:358</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.3389/fmicb.2015.01090"></div>
<div class="__dimensions_badge_embed__" data-doi="10.3389/fmicb.2015.01090" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.3389/fmicb.2015.01090" target="_new">Genome reconstructions indicate the partitioning of ecological functions inside a phytoplankton bloom in the Amundsen Sea, Antarctica.</a></h3>
    <span class="pub-authors"><span class="pub-member-author">Delmont, T. O.</span>, <span class="pub-member-author">Eren, A. M</span>., Vineis, J. H., and Post, A. F.</span>
    <span class="pub-journal"><i>Front Microbiol</i>. <b>6:1090</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1128/AEM.01524-15"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1128/AEM.01524-15" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1128/AEM.01524-15" target="_new">Comparison of sewage and animal fecal microbiomes using oligotyping reveals potential human fecal indicators in multiple taxonomic groups.</a></h3>
    <span class="pub-authors">Fisher, J. C., <span class="pub-member-author">Eren, A. M</span>., Green, H. C., Shanks, O. C., Morrison, H. G., Vineis, J. H., Sogin, M. L., and McLellan, S. L.</span>
    <span class="pub-journal"><i>Appl Environ Microbiol</i>. <b>81, 7023–7033</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.3389/fmicb.2015.00358"></div>
<div class="__dimensions_badge_embed__" data-doi="10.3389/fmicb.2015.00358" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.3389/fmicb.2015.00358" target="_new">Reconstructing Rare Soil Microbial Genomes using in situ Enrichments and Metagenomics.</a></h3>
    <span class="pub-authors"><span class="pub-member-author">Delmont, T. O.</span>, <span class="pub-member-author">Eren, A. M</span>., Maccario, L., Prestat, E., <span class="pub-member-author">Esen, Ö. C.</span>, Pelletier, E., LePaslier, D., Simonet, P., and Vogel, T.</span>
    <div class="pub-info">
    <div class="pub-featured-image">
    <a href="../images/pubs/delmont_rare_soil_genomes.png"><img src="../images/pubs/delmont_rare_soil_genomes.png" style="max-width: 100px; max-height: 80px; width: auto; border: none; height: auto; margin: 0 auto; display: block; transform: translateY(15%);"/></a>
    </div>
    <div class="pub-highlights">
    <span style="display: inline-block; padding-bottom: 5px;">-  Application of <i>in situ</i> enrichments to shape the diversity of complex metagenomes prior to sequencing.</span><br><span style="display: inline-block; padding-bottom: 5px;">-  The first successful attempt to assemble multiple near-complete bacterial genomes directly from a soil sample.</span><br><span style="display: inline-block; padding-bottom: 5px;">-  The abundance of reconstructed genomes ranged from rare (<0.0001%) to relatively abundant (>0.01%) in pristine soil metagenomes.</span>
    </div>
    </div>
    <span class="pub-journal"><i>Front Microbiol</i>. <b>6:358</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1038/ismej.2014.97"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1038/ismej.2014.97" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1038/ismej.2014.97" target="_new">A single genus in the gut microbiome reflects host preference and specificity.</a></h3>
    <span class="pub-authors"><span class="pub-member-author">Eren, A. M</span>., Sogin, M. L., Morrison, H. G., Vineis, J. H., Fisher, J. C., Newton, R. J., and McLellan, S. L.</span>
    <span class="pub-journal"><i>ISME J</i>. <b>9(1), 90-100</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1038/ismej.2014.195"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1038/ismej.2014.195" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1038/ismej.2014.195" target="_new">Minimum entropy decomposition: Unsupervised oligotyping for sensitive partitioning of high-throughput marker gene sequences.</a></h3>
    <span class="pub-authors"><span class="pub-member-author">Eren, A. M</span>., Morrison, H. G., Lescault, P. J., Reveillaud, J., Vineis, J. H., and Sogin, M. L.</span>
    <span class="pub-journal"><i>ISME J</i>. <b>9, 968-79</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1128/mBio.02574-14"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1128/mBio.02574-14" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1128/mBio.02574-14" target="_new">Sewage reflects the microbiomes of human populations.</a></h3>
    <span class="pub-authors">Newton, R. J., McLellan, S. L., Dila, D. K., Vineis, J. H., Morrison, H. G., <span class="pub-member-author">Eren, A. M</span>., and Sogin, M. L.</span>
    <span class="pub-journal"><i>mBio</i>. <b>6:2</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1038/ajg.2015.357"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1038/ajg.2015.357" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1038/ajg.2015.357" target="_new">Extensive Modulation of the Fecal Metagenome in Children With Crohn's Disease During Exclusive Enteral Nutrition.</a></h3>
    <span class="pub-authors">Quince, C., Ijaz, U. Z., Loman, N., <span class="pub-member-author">Eren, A. M</span>., Saulnier, D., Russell, J., Haig, S. J., Calus, S. T., Quick, J., Barclay, A., Bertz, M., Hansen, R., McGrogan, P., Russel, K. R., Edwards, C. A., and Gerasimidis, K.</span>
    <span class="pub-journal"><i>Am J Gastroenterol</i>. <b>110:1718–1729</b></span>
</div>


<a name="2014">&nbsp;</a>
<h1>2014</h1>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1073/pnas.1409644111"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1073/pnas.1409644111" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1073/pnas.1409644111" target="_new">Oligotyping analysis of the human oral microbiome.</a></h3>
    <span class="pub-authors"><span class="pub-member-author">Eren, A. M</span>., Borisy, G. G., Huse, S. M., and Mark Welch, J. L.</span>
    <span class="pub-journal"><i>Proc Natl Acad Sci U S A</i>. <b>111(28), E2875-84</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1093/bib/bbt010"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1093/bib/bbt010" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1093/bib/bbt010" target="_new">DRISEE overestimates errors in metagenomic sequencing data.</a></h3>
    <span class="pub-authors"><span class="pub-member-author">Eren, A. M</span>., Morrison, H. G., Huse, S. M., and Sogin, M. L.</span>
    <span class="pub-journal"><i>Brief Bioinform</i>. <b>15(5), 783-7</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1371/journal.pone.0090785"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1371/journal.pone.0090785" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1371/journal.pone.0090785" target="_new">Humpback whale populations share a core skin bacterial community: towards a health index for marine mammals?</a></h3>
    <span class="pub-authors">Apprill, A., Robbins, J., <span class="pub-member-author">Eren, A. M</span>., Pack, A. A., Reveillaud, J., Mattila, D., Moore, M., Niemeyer, M., Moore, K. M., and Mincer, T. J.</span>
    <span class="pub-journal"><i>PLoS One</i>. <b>9(3), e90785</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1128/mBio.00682-13"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1128/mBio.00682-13" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1128/mBio.00682-13" target="_new">Ecological succession and stochastic variation in the assembly of Arabidopsis thaliana phyllosphere communities.</a></h3>
    <span class="pub-authors">Maignien, L., DeForce, E. A., Chafee, M. E., <span class="pub-member-author">Eren, A. M</span>., and Simmons, S. L.</span>
    <span class="pub-journal"><i>MBio</i>. <b>5(1), e00682-13</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1093/icb/icu065"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1093/icb/icu065" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1093/icb/icu065" target="_new">SeaBase: a multispecies transcriptomic resource and platform for gene network inference.</a></h3>
    <span class="pub-authors">Fischer, A. H., Mozzherin, D., <span class="pub-member-author">Eren, A. M</span>., Lans, K. D., Wilson, N., Cosentino, C., and Smith, J.</span>
    <span class="pub-journal"><i>Integr Comp Biol</i>. <b>54(2), 250-63</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1186/1471-2105-15-41"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1186/1471-2105-15-41" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1186/1471-2105-15-41" target="_new">VAMPS: a website for visualization and analysis of microbial population structures.</a></h3>
    <span class="pub-authors">Huse, S. M., Mark Welch, D. B., Voorhis, A., Shipunova, A., Morrison, H. G., <span class="pub-member-author">Eren, A. M</span>., and Sogin, M. L.</span>
    <span class="pub-journal"><i>BMC Bioinformatics</i>. <b>15(1), 41</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1111/1758-2229.12189"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1111/1758-2229.12189" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1111/1758-2229.12189" target="_new">Blautia and Prevotella sequences distinguish human and animal fecal pollution in Brazil surface waters.</a></h3>
    <span class="pub-authors">Koskey, A. M., Fisher, J. C., <span class="pub-member-author">Eren, A. M</span>., Ponce-Terashima, R., Reis, M. G., Blanton, R. E., and McLellan, S. L.</span>
    <span class="pub-journal"><i>Environ Microbiol Rep</i>. <b>6(6), 696-704</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1016/j.tim.2014.08.002"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1016/j.tim.2014.08.002" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1016/j.tim.2014.08.002" target="_new">Discovering new indicators of fecal pollution.</a></h3>
    <span class="pub-authors">McLellan, S. L., and <span class="pub-member-author">Eren, A. M</span>.</span>
    <span class="pub-journal"><i>Trends Microbiol</i>. <b>22(12), 697-706</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1038/ismej.2013.227"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1038/ismej.2013.227" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1038/ismej.2013.227" target="_new">Host-specificity among abundant and rare taxa in the sponge microbiome.</a></h3>
    <span class="pub-authors">Reveillaud, J., Maignien, L., <span class="pub-member-author">Eren, A. M</span>., Huber, J. A., Apprill, A., Sogin, M. L., and Vanreusel, A.</span>
    <span class="pub-journal"><i>ISME J</i>. <b>8(6), 1198-209</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.3389/fmicb.2014.00568"></div>
<div class="__dimensions_badge_embed__" data-doi="10.3389/fmicb.2014.00568" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.3389/fmicb.2014.00568" target="_new">Dynamics of tongue microbial communities with single-nucleotide resolution using oligotyping.</a></h3>
    <span class="pub-authors">Mark Welch, J. L., Utter, D. R., Rossetti, B. J., Mark Welch, D. B., <span class="pub-member-author">Eren, A. M</span>., and Borisy, G. G.</span>
    <span class="pub-journal"><i>Front Microbiol</i>. <b>5:568</b></span>
</div>


<a name="2013">&nbsp;</a>
<h1>2013</h1>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1111/2041-210X.12114"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1111/2041-210X.12114" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1111/2041-210X.12114" target="_new">Oligotyping: Differentiating between closely related microbial taxa using 16S rRNA gene data.</a></h3>
    <span class="pub-authors"><span class="pub-member-author">Eren, A. M</span>., Maignien, L., Sul, W. J., Murphy, L. G., Grim, S. L., Morrison, H. G., and Sogin, M. L.</span>
    <span class="pub-journal"><i>Methods Ecol Evol</i>. <b>4(12)</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1371/journal.pone.0066643"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1371/journal.pone.0066643" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1371/journal.pone.0066643" target="_new">A filtering method to generate high quality short reads using illumina paired-end technology.</a></h3>
    <span class="pub-authors"><span class="pub-member-author">Eren, A. M</span>., Vineis, J. H., Morrison, H. G., and Sogin, M. L.</span>
    <span class="pub-journal"><i>PLoS One</i>. <b>8(6), e66643</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1097/MPG.0b013e318282a8c2"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1097/MPG.0b013e318282a8c2" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1097/MPG.0b013e318282a8c2" target="_new">Histamine-2 receptor blockers alter the fecal microbiota in premature infants.</a></h3>
    <span class="pub-authors">Gupta, R. W., Tran, L., Norori, J., Ferris, M. J., <span class="pub-member-author">Eren, A. M</span>., Taylor, C. M., Dowd, S. E., and Penn, D.</span>
    <span class="pub-journal"><i>J Pediatr Gastroenterol Nutr</i>. <b>56(4), 397-400</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1111/1462-2920.12092"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1111/1462-2920.12092" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1111/1462-2920.12092" target="_new">Sewage reflects the distribution of human faecal Lachnospiraceae.</a></h3>
    <span class="pub-authors">McLellan, S. L., Newton, R. J., Vandewalle, J. L., Shanks, O. C., Huse, S. M., <span class="pub-member-author">Eren, A. M</span>., and Sogin, M. L.</span>
    <span class="pub-journal"><i>Environ Microbiol</i>. <b>15(8), 2213-27</b></span>
</div>

<div class="pub">
<div class='altmetric-embed' data-badge-type='donut' data-doi="10.1186/2049-2618-1-9"></div>
<div class="__dimensions_badge_embed__" data-doi="10.1186/2049-2618-1-9" data-hide-zero-citations="true" data-legend="hover-bottom" data-style="small_circle"></div>
    <h3><a href=" https://doi.org/10.1186/2049-2618-1-9" target="_new">Multiphasic analysis of the temporal development of the distal gut microbiota in patients following ileal pouch anal anastomosis.</a></h3>
    <span class="pub-authors">Young, V. B., Raffals, L. H., Huse, S. M., Vital, M., Dai, D., Schloss, P. D., Brulc, J. M., Antonopoulos, D. A., Arrieta, R. L., Kwon, J. H., Reddy, K. G., Hubert, N. A., Grim, S. L., Vineis, J. H., Dalal, S., Morrison, H. G., <span class="pub-member-author">Eren, A. M</span>., Meyer, F., Schmidt, T. M., Tiedje, J. M., Chang, E. B., and Sogin, M. L.</span>
    <span class="pub-journal"><i>Microbiome</i>. <b>1(1), 9</b></span>
</div>


